# *ShiNyP* {#sec-shinyp}

::: {.highlighted-text style="padding-top: 15px; padding-bottom: 15px; padding-left: 15px"}
| ðŸ“„ **Input data:** Genome-wide biallelic SNP in Variant Call Format (VCF) file format.
| ðŸ“Š **Analysis:** Data QC, population genetics analysis, core collectionâ€¦
| ðŸ“‹ **Output:** Publication-ready figures, tables, analyzed data objects, and AI-driven report.
:::

------------------------------------------------------------------------

| **Key Features**
| Real-time Processing, Analysis, and Visualization of SNP Datasets:
| \> Comprehensive statistical and computational exploration
| \> Customizable visualization options
| \> Publication-ready figures and tables
| \> Reproducible analyzed data objects
| \> AI-driven report generation

#### Publication {.unnumbered}

Huang et al. (upcoming 2024) *ShiNyP*: An Interactive Shiny-Based Platform for Genome-Wide SNP Analysis and Visualization

#### Support {.unnumbered}

If you encounter any issues or have suggestions for new features, please submit a report through our feedback form: <a href="https://forms.gle/GPCggSo5czyNLfoB7" target="_blank">https://forms.gle/GPCggSo5czyNLfoB7</a>

#### Websites {.unnumbered}

-   **Journal Article**: <a href="https://www.example.com" target="_blank">Under Review...</a>

-   **GitHub Repository**: <a href="https://github.com/TeddYenn/ShiNyP" target="_blank">https://github.com/TeddYenn/ShiNyP</a>

-   **User Manual**: <a href="https://github.com/TeddYenn/ShiNyP" target="_blank">https://teddyenn.github.io/ShiNyP-guide</a>

-   **Demo Datasets**: <a href="https://reurl.cc/QEx5lZ" target="_blank">https://reurl.cc/QEx5lZ</a>

-   **Demo Platform**: <a href="https://teddyhuang.shinyapps.io/ShiNyP/" target="_blank">https://teddyhuang.shinyapps.io/ShiNyP</a>

-   **Feedback Form**: <a href="https://forms.gle/GPCggSo5czyNLfoB7" target="_blank">https://forms.gle/GPCggSo5czyNLfoB7</a>

-   **Podcast Program**: <a href="https://notebooklm.google.com/notebook/a952a309-6372-428d-90af-71593d784dd1/audio"target="_blank">https://reurl.cc/WN6R35</a>

------------------------------------------------------------------------

![](images/Fig.%201-02.jpg)

| â–² **An overview of the [*ShiNyP*]{style="color: #9D6853;"} platform's workflow for genome-wide SNP data analysis.**
| - Data Input & Processing: Beginning with <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html" target="_blank">VCF data input</a>, it performs <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html" target="_blank">quality control (QC)</a> and <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html" target="_blank">data transformation</a> steps.
| - Analysis & Output: The analysis phase is divided into modules, each represented as a page in the platform, with multiple subpages offering specific analytical functions, including <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html" target="_blank">population structure</a>, <a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html" target="_blank">genetic diversity</a>, <a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html" target="_blank">selection sweep</a>, and <a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html" target="_blank">core collection</a>.
| - Customized Output: The final output offers tailored visualizations and includes an AI-generated report summarizing the results. The pipeline streamlines data input, processing, and advanced analysis to deliver publication-ready figures and reports customized to the userâ€™s needs.
| \*Subpage frame colors indicate available functions for customization. For example, blue frames for [PCA]{.underline} and [DAPC]{.underline} correspond to the Scatter Plot ^Plus^ tool for customizing scatter plots, while red and purple frames correspond to Tree Plot ^Plus^ and Manhattan Plot ^Plus^, respectively.
