<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6 Population Structure | ShiNyP: User Guide</title>
  <meta name="description" content="A Real-Time Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6 Population Structure | ShiNyP: User Guide" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A Real-Time Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  <meta name="github-repo" content="TeddYenn/ShiNyP" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6 Population Structure | ShiNyP: User Guide" />
  
  <meta name="twitter:description" content="A Real-Time Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sec-data-conversion.html"/>
<link rel="next" href="sec-genetic-diversity.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ShiNyP: User Guide</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Quickstart</a></li>
<li class="chapter" data-level="2" data-path="sec-shinyp.html"><a href="sec-shinyp.html"><i class="fa fa-check"></i><b>2</b> <em>ShiNyP</em></a></li>
<li class="chapter" data-level="3" data-path="sec-data-input.html"><a href="sec-data-input.html"><i class="fa fa-check"></i><b>3</b> Data Input</a>
<ul>
<li class="chapter" data-level="3.1" data-path="sec-data-input.html"><a href="sec-data-input.html#vcf"><i class="fa fa-check"></i><b>3.1</b> VCF</a></li>
<li class="chapter" data-level="3.2" data-path="sec-data-input.html"><a href="sec-data-input.html#data.framegenindgenlight"><i class="fa fa-check"></i><b>3.2</b> data.frame/genind/genlight</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sec-data-qc.html"><a href="sec-data-qc.html"><i class="fa fa-check"></i><b>4</b> Data QC</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sec-data-qc.html"><a href="sec-data-qc.html#sample-qc"><i class="fa fa-check"></i><b>4.1</b> Sample QC</a></li>
<li class="chapter" data-level="4.2" data-path="sec-data-qc.html"><a href="sec-data-qc.html#snp-qc"><i class="fa fa-check"></i><b>4.2</b> SNP QC</a></li>
<li class="chapter" data-level="4.3" data-path="sec-data-qc.html"><a href="sec-data-qc.html#snp-density"><i class="fa fa-check"></i><b>4.3</b> SNP Density</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sec-data-conversion.html"><a href="sec-data-conversion.html"><i class="fa fa-check"></i><b>5</b> Data Transform</a></li>
<li class="chapter" data-level="6" data-path="sec-population-structure.html"><a href="sec-population-structure.html"><i class="fa fa-check"></i><b>6</b> Population Structure</a>
<ul>
<li class="chapter" data-level="6.1" data-path="sec-population-structure.html"><a href="sec-population-structure.html#pca-principal-component-analysis"><i class="fa fa-check"></i><b>6.1</b> PCA (Principal Component Analysis)</a></li>
<li class="chapter" data-level="6.2" data-path="sec-population-structure.html"><a href="sec-population-structure.html#dapc-discriminant-analysis-of-principal-components"><i class="fa fa-check"></i><b>6.2</b> DAPC (Discriminant Analysis of Principal Components)</a></li>
<li class="chapter" data-level="6.3" data-path="sec-population-structure.html"><a href="sec-population-structure.html#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree"><i class="fa fa-check"></i><b>6.3</b> UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree</a></li>
<li class="chapter" data-level="6.4" data-path="sec-population-structure.html"><a href="sec-population-structure.html#nj-neighbor-joining-tree"><i class="fa fa-check"></i><b>6.4</b> NJ (Neighbor-Joining) Tree</a></li>
<li class="chapter" data-level="6.5" data-path="sec-population-structure.html"><a href="sec-population-structure.html#kinship-analysis"><i class="fa fa-check"></i><b>6.5</b> Kinship Analysis</a></li>
<li class="chapter" data-level="6.6" data-path="sec-population-structure.html"><a href="sec-population-structure.html#scatter-plot-plus"><i class="fa fa-check"></i><b>6.6</b> Scatter Plot <sup>Plus</sup></a></li>
<li class="chapter" data-level="6.7" data-path="sec-population-structure.html"><a href="sec-population-structure.html#tree-plot-plus"><i class="fa fa-check"></i><b>6.7</b> Tree Plot <sup>Plus</sup></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html"><i class="fa fa-check"></i><b>7</b> Genetic Diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#diversity-parameter"><i class="fa fa-check"></i><b>7.1</b> Diversity Parameter</a></li>
<li class="chapter" data-level="7.2" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#circos-plot"><i class="fa fa-check"></i><b>7.2</b> Circos Plot</a></li>
<li class="chapter" data-level="7.3" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#genetic-distance"><i class="fa fa-check"></i><b>7.3</b> Genetic Distance</a></li>
<li class="chapter" data-level="7.4" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#amova-analysis-of-molecular-variance"><i class="fa fa-check"></i><b>7.4</b> AMOVA (Analysis of MOlecular VAriance)</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html"><i class="fa fa-check"></i><b>8</b> Selection Sweep</a>
<ul>
<li class="chapter" data-level="8.1" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#pcadapt"><i class="fa fa-check"></i><b>8.1</b> pcadapt</a></li>
<li class="chapter" data-level="8.2" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#outflank"><i class="fa fa-check"></i><b>8.2</b> OutFLANK</a></li>
<li class="chapter" data-level="8.3" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#ibs-identity-by-state"><i class="fa fa-check"></i><b>8.3</b> IBS (Identity By State)</a></li>
<li class="chapter" data-level="8.4" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#manhattan-plot-plus"><i class="fa fa-check"></i><b>8.4</b> Manhattan Plot <sup>Plus</sup></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="sec-core-collection.html"><a href="sec-core-collection.html"><i class="fa fa-check"></i><b>9</b> Core Collection</a>
<ul>
<li class="chapter" data-level="9.1" data-path="sec-core-collection.html"><a href="sec-core-collection.html#core-sample-set"><i class="fa fa-check"></i><b>9.1</b> Core Sample Set</a></li>
<li class="chapter" data-level="9.2" data-path="sec-core-collection.html"><a href="sec-core-collection.html#core-snp-set"><i class="fa fa-check"></i><b>9.2</b> Core SNP Set</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sec-ai-report.html"><a href="sec-ai-report.html"><i class="fa fa-check"></i><b>10</b> AI Report</a></li>
<li class="chapter" data-level="11" data-path="sec-index.html"><a href="sec-index.html"><i class="fa fa-check"></i><b>11</b> INDEX</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ShiNyP: User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sec-population-structure" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">6</span> Population Structure<a href="sec-population-structure.html#sec-population-structure" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="highlighted-text" style="padding-top: 15px; padding-bottom: 1px; padding-left: 15px">
<p>➡️ This section contains seven subpages: <u><strong>PCA</strong></u>, <u><strong>DAPC</strong></u>, <u><strong>UPGMA Tree</strong></u>, <u><strong>NJ Tree</strong></u>, <u><strong>Kinship</strong></u>, <u><strong>Scatter Plot</strong></u><sup><strong>Plus</strong></sup>, and <u><strong>Tree Plot</strong></u><sup><strong>Plus</strong></sup> allowing you to conduct various population structure analyses and customize your plot.</p>
</div>
<p><img src="images/Supp.%20Fig.%201-5_%E9%A0%81%E9%9D%A2_2.jpg" /></p>
<div id="pca-principal-component-analysis" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> PCA (Principal Component Analysis)<a href="sec-population-structure.html#pca-principal-component-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A widely used method to uncover underlying population structure by reducing the dimensionality of genetic data.</p>
<div id="required-dataset-1" class="section level5 unnumbered hasAnchor">
<h5>Required Dataset:<a href="sec-population-structure.html#required-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
</ul>
</div>
<div id="one-step-1" class="section level5 unnumbered hasAnchor">
<h5><strong>One Step:</strong><a href="sec-population-structure.html#one-step-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run PCA</strong></span> button to generate PCA plots and the following downloadable files.</li>
</ol>
<blockquote>
<p><strong>Note</strong>: You can upload the <strong>Group Info.</strong> (from <u>Population Structure</u>/<u>DAPC</u>) or <strong>Core Sample Info.</strong> (from <u>Core Collection</u>/<u>Core Sample Set</u>) to classify individuals and color them in the PCA Scatter Plot.</p>
</blockquote>
</div>
<div id="outputs-5" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-population-structure.html#outputs-5" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>PCA Scatter Plot (PDF)</strong>: A scatter plot showing the distribution of samples based on principal components, with each dot representing an individual.</p></li>
<li><p><strong>PC Explained Variance Plot (PDF)</strong>: Visualizes the variance explained by each principal component.</p></li>
<li><p><strong>Explained Variance (CSV)</strong>: Contains the explained variance of each principal component.</p></li>
<li><p><strong>PCA Transformed Data (CSV)</strong>: Dataset transformed into principal components, with samples as rows and principal components as columns.</p></li>
<li><p><strong>PCA Object (RDS)</strong>: Contains all PCA results for future use and reproducibility, and can be used as input data in the <u>Population Structure</u>/<u>Scatter Plot</u><sup>Plus</sup> subpage.</p></li>
</ul>
<p><img src="images/clipboard-534056671.png" /></p>
<p><em>PCA Complete!</em></p>
</div>
</div>
<div id="dapc-discriminant-analysis-of-principal-components" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> DAPC (Discriminant Analysis of Principal Components)<a href="sec-population-structure.html#dapc-discriminant-analysis-of-principal-components" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A multivariate method for identifying and visualizing genetic clusters by combining PCA and Linear Discriminant Analysis (LDA) <span class="citation">(<a href="#ref-jombart2010">Jombart, Devillard, and Balloux 2010</a>)</span>. For more information, visit <a href="https://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf" class="uri">https://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf</a>.</p>
<div id="required-dataset-2" class="section level5 unnumbered hasAnchor">
<h5>Required Dataset:<a href="sec-population-structure.html#required-dataset-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>genind</code></strong></span></li>
</ul>
</div>
<div id="step-1-cluster-identification" class="section level5 unnumbered hasAnchor">
<h5>Step 1: <strong>Cluster Identification</strong><a href="sec-population-structure.html#step-1-cluster-identification" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run DAPC I</strong></span> button to determine the optimal number of clusters (the lowest BIC value indicates the optimal number of clusters).</li>
</ol>
<blockquote>
<p><strong>Note:</strong> The default number of PC axes for cluster identification is set to retain PCs that capture up to 80% of the total variance. You can refer the “PC Explained Variance Plot” in the <u>Population Structure</u>/<u>PCA</u> subpage.</p>
</blockquote>
</div>
<div id="step-2-dapc-analysis" class="section level5 unnumbered hasAnchor">
<h5>Step 2: <strong>DAPC Analysis</strong><a href="sec-population-structure.html#step-2-dapc-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p>Choose the number of cluster (K) based on the “Bayesian Information Criterion (BIC) Plot”.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run DAPC II</strong></span> button to generate DAPC plots and the following downloadable files.</p></li>
</ol>
<blockquote>
<p><strong>Note</strong>: You can download the “DAPC Object” and upload it on <u>Population Structure</u>/<u>Scatter Plot</u><sup>Plus</sup> subpage to customize your 2D and 3D scatter plots.</p>
</blockquote>
</div>
<div id="outputs-6" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-population-structure.html#outputs-6" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>Bayesian Information Criterion (BIC) Plot (PDF)</strong>: Visual representation of the BIC for model selection.</p></li>
<li><p><strong>Density Plot of First &amp; Second Discriminant Function</strong> <strong>(PDF)</strong>: Displays the density of the first and second discriminant functions, with each row bar representing an individual.</p></li>
<li><p><strong>DAPC Scatter Plot (PDF)</strong>: A scatter plot showing the distribution of samples based on discriminant functions (x-axis: first discriminant function; y-axis: second discriminant function), with each dot representing an individual.</p></li>
<li><p><strong>DAPC Membership Probability Plot (PDF)</strong>: Visualizes membership probabilities of individuals in different groups, with each row bar representing an individual.</p></li>
<li><p><strong>DAPC Group Info. (CSV)</strong>: Contains the group assignments for each individual based on DAPC. This file used in various subpages.</p></li>
<li><p><strong>DAPC Transformed Data (CSV)</strong>: Dataset transformed into discriminant functions with samples as rows and discriminant functions as columns.</p></li>
<li><p><strong>DAPC Object (RDS)</strong>: Contains all results from the DAPC analysis for future reproducibility. It can be used as input data in the <u>Population Structure</u>/<u>Scatter Plot</u><sup>Plus</sup> and <u>Core Collection</u>/<u>Core SNP Set</u> subpages.</p></li>
</ul>
<p><img src="images/clipboard-1914132278.png" /></p>
<p><em>DAPC Complete!</em></p>
</div>
</div>
<div id="upgma-unweighted-pair-group-method-with-arithmetic-mean-tree" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree<a href="sec-population-structure.html#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A classic approaach for constructing rooted trees based on genetic distance data. UPGMA tree is generated by <em>poppr</em> and <em>ggtree</em> packages <span class="citation">(<a href="#ref-yu2016">Yu et al. 2016</a>; <a href="#ref-kamvar2014">Kamvar, Tabima, and Grünwald 2014</a>)</span>.</p>
<div id="required-dataset-3" class="section level5 unnumbered hasAnchor">
<h5>Required Dataset:<a href="sec-population-structure.html#required-dataset-3" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>genlight</code></strong></span></li>
</ul>
</div>
<div id="steps-1" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps</strong>:<a href="sec-population-structure.html#steps-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p>Choose the number of bootstrap replicates, which will be used for assessing the confidence of the branching structure.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run UPGMA</strong></span> button to generate tree plot.</p></li>
</ol>
<blockquote>
<p><strong>Note</strong>: You can download the “UPGMA Object” and upload it on <u>Population Structure</u>/<u>Tree Plot</u><sup>Plus</sup> subpage to customize your phylogenetic tree.</p>
</blockquote>
</div>
<div id="outputs-7" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-population-structure.html#outputs-7" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>UPGMA Phylogenetic Tree (PDF)</strong>: A UPGMA rooted tree with a user-defined layout style.</p></li>
<li><p><strong>UPGMA Object (RDS)</strong>: Contains all information of the UPGMA tree, and can be used as input data in the <u>Population Structure</u>/<u>Tree Plot</u><sup>Plus</sup> subpage.</p></li>
</ul>
<p><img src="images/clipboard-3531175086.png" /></p>
<p><em>UPGMA Tree Complete!</em></p>
</div>
</div>
<div id="nj-neighbor-joining-tree" class="section level2 hasAnchor" number="6.4">
<h2><span class="header-section-number">6.4</span> NJ (Neighbor-Joining) Tree<a href="sec-population-structure.html#nj-neighbor-joining-tree" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A method for building unrooted trees using genetic distance data. NJ tree is generated by <em>ape</em> and <em>ggtree</em> packages <span class="citation">(<a href="#ref-paradis2018">Paradis and Schliep 2018</a>; <a href="#ref-yu2016">Yu et al. 2016</a>)</span>.</p>
<div id="required-dataset-4" class="section level5 unnumbered hasAnchor">
<h5>Required Dataset:<a href="sec-population-structure.html#required-dataset-4" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>genlight</code></strong></span></li>
</ul>
</div>
<div id="one-step-2" class="section level5 unnumbered hasAnchor">
<h5><strong>One Step:</strong><a href="sec-population-structure.html#one-step-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run NJ</strong></span> button to generate tree plot.</li>
</ol>
<blockquote>
<p><strong>Note</strong>: You can download the “NJ Object” and upload it on <u>Population Structure</u>/<u>Tree Plot</u><sup>Plus</sup> subpage to customize your phylogenetic tree.</p>
</blockquote>
</div>
<div id="outputs-8" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-population-structure.html#outputs-8" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>NJ Phylogenetic Tree (PDF)</strong>: A NJ unrooted tree with a user-defined layout style.</p></li>
<li><p><strong>NJ Object (RDS)</strong>: Contains all information of the NJ tree, and can be used as input data in the <u>Population Structure</u>/<u>Tree Plot</u><sup>Plus</sup> subpage.</p></li>
</ul>
<p><img src="images/clipboard-1096590194.png" /></p>
<p><em>NJ Tree Complete!</em></p>
</div>
</div>
<div id="kinship-analysis" class="section level2 hasAnchor" number="6.5">
<h2><span class="header-section-number">6.5</span> Kinship Analysis<a href="sec-population-structure.html#kinship-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A statistical method for assessing genetic relationships and relatedness among individuals based on shared alleles <span class="citation">(<a href="#ref-kang2010">Kang et al. 2010</a>)</span>. Kinship matrix is generated by <em>statgenGWAS</em> package.For more information, visit <a href="https://rdrr.io/cran/statgenGWAS/man/kinship.html" class="uri">https://rdrr.io/cran/statgenGWAS/man/kinship.html</a>.</p>
<div id="required-dataset-5" class="section level5 unnumbered hasAnchor">
<h5>Required Dataset:<a href="sec-population-structure.html#required-dataset-5" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
</ul>
</div>
<div id="steps-2" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-population-structure.html#steps-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Group Info.</strong> from <u>Population Structure</u>/<u>DAPC</u> (optional). If uploaded, the order of samples will follow the group assignment; otherwise, it will follow the order of the original VCF data.</p></li>
<li><p>Choose a method to run kinship analysis.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Kinship</strong></span> button to generate the kinship matrix.</p></li>
</ol>
</div>
<div id="outputs-9" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-population-structure.html#outputs-9" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>Kinship Matrix Plot (PDF)</strong>: A visual representation of the kinship matrix.</p></li>
<li><p><strong>Kinship Matrix (RDS)</strong>: Contains the kinship matrix data.</p></li>
</ul>
<blockquote>
<p>Note: This kinship matrix can be directly used as input for <em>GAPIT</em> package in genome-wide association studies (GWAS), helping to control for confounding effects.</p>
</blockquote>
<p><img src="images/clipboard-2215706632.png" /></p>
<p><em>Kinship Analysis Complete!</em></p>
</div>
</div>
<div id="scatter-plot-plus" class="section level2 hasAnchor" number="6.6">
<h2><span class="header-section-number">6.6</span> Scatter Plot <sup>Plus</sup><a href="sec-population-structure.html#scatter-plot-plus" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Customize your scatter plot based on the results from <u>Population Structure</u>/<u>PCA</u> or <u>Population Structure</u>/<u>DAPC</u>.</p>
<div id="required-files" class="section level5 unnumbered hasAnchor">
<h5>Required Files:<a href="sec-population-structure.html#required-files" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><strong>PCA Object</strong> (PCA_prcomp_Object.rds file) or <strong>DAPC Object</strong> (DAPC_dapc_Object.rds file)</li>
<li><strong>Group and Other Info.</strong> (modifiable from DAPC_Group_Info.csv)</li>
</ul>
<blockquote>
<p><strong>Note</strong>: You can add more information about samples by adding new variables to the Group Info. file. Ensure that the sample order remains unchanged.</p>
</blockquote>
<p>▼ Example of Group Info. file (CSV).</p>
<p><img src="images/clipboard-4028088589.png" width="350" /></p>
</div>
<div id="section" class="section level5 unnumbered">
<h5 class="unnumbered"></h5>
</div>
<div id="steps-3" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-population-structure.html#steps-3" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>PCA or DAPC Object (RDS)</strong></p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Group and Other Info. (CSV)</strong></p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Scatter Plot</strong></span> button to generate the 2D and 3D interactive scatter plots.</p></li>
<li><p>Customize the scatter plot and click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Scatter Plot</strong></span> button again.</p></li>
</ol>
<blockquote>
<p><strong>Note</strong>: The scatter plots are downloaded as HTML files and can be opened with browsers like Chrome or Edge.</p>
</blockquote>
</div>
<div id="outputs-10" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-population-structure.html#outputs-10" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>2D Scatter Plot (HTML)</strong>: Two-dimensional interactive scatter plot with user-defined attributes.</p></li>
<li><p><strong>3D Scatter Plot (HTML)</strong>: Three-dimensional interactive scatter plot with user-defined attributes.</p></li>
</ul>
<p><img src="images/clipboard-1525634301.png" /></p>
<p><em>Scatter Plot <sup>Plus</sup> Complete!</em></p>
</div>
</div>
<div id="tree-plot-plus" class="section level2 hasAnchor" number="6.7">
<h2><span class="header-section-number">6.7</span> Tree Plot <sup>Plus</sup><a href="sec-population-structure.html#tree-plot-plus" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Customize your phylogenetic tree plot based on the results from <u>Population Structure</u>/<u>UPGMA</u> or <u>Population Structure</u>/<u>NJ</u>.</p>
<div id="required-files-1" class="section level5 unnumbered hasAnchor">
<h5>Required Files:<a href="sec-population-structure.html#required-files-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><strong>UPGMA Object</strong> (UPGMA_phylo_Object.rds file) or <strong>NJ Object</strong> (NJ_phylo_Object.rds file)</li>
<li><strong>Group and Other Info.</strong> (modifiable from DAPC_Group_Info.csv)</li>
</ul>
<blockquote>
<p><strong>Note</strong>: You can add more information about samples by adding new variables to the Group Info. file. Ensure that the sample order remains unchanged.</p>
</blockquote>
</div>
<div id="steps-4" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-population-structure.html#steps-4" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>UPGMA or NJ Object (RDS)</strong></p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Group and Other Info. (CSV)</strong></p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Tree Plot</strong></span> button to generate the tree plot.</p></li>
<li><p>Customize the tree plot and click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Tree Plot</strong></span> button again.</p></li>
</ol>
</div>
<div id="outputs-11" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-population-structure.html#outputs-11" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><strong>Phylogenetic Tree Plot (PDF)</strong>: A phylogenetic tree plot with user-defined layout style and attributes.</li>
</ul>
<p><img src="images/clipboard-2211908142.png" /></p>
<p><em>Tree Plot <sup>Plus</sup> Complete!</em></p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-jombart2010" class="csl-entry">
Jombart, Thibaut, Sébastien Devillard, and François Balloux. 2010. <span>“Discriminant Analysis of Principal Components: A New Method for the Analysis of Genetically Structured Populations.”</span> <em>BMC Genetics</em> 11 (1): 94. <a href="https://doi.org/10.1186/1471-2156-11-94">https://doi.org/10.1186/1471-2156-11-94</a>.
</div>
<div id="ref-kamvar2014" class="csl-entry">
Kamvar, Zhian N., Javier F. Tabima, and Niklaus J. Grünwald. 2014. <span>“<span><em>Poppr</em></span>: An R Package for Genetic Analysis of Populations with Clonal, Partially Clonal, and/or Sexual Reproduction.”</span> <em>PeerJ</em> 2 (March): e281. <a href="https://doi.org/10.7717/peerj.281">https://doi.org/10.7717/peerj.281</a>.
</div>
<div id="ref-kang2010" class="csl-entry">
Kang, Hyun Min, Jae Hoon Sul, Susan K Service, Noah A Zaitlen, Sit-yee Kong, Nelson B Freimer, Chiara Sabatti, and Eleazar Eskin. 2010. <span>“Variance Component Model to Account for Sample Structure in Genome-Wide Association Studies.”</span> <em>Nature Genetics</em> 42 (4): 348–54. <a href="https://doi.org/10.1038/ng.548">https://doi.org/10.1038/ng.548</a>.
</div>
<div id="ref-paradis2018" class="csl-entry">
Paradis, Emmanuel, and Klaus Schliep. 2018. <span>“Ape 5.0: An Environment for Modern Phylogenetics and Evolutionary Analyses in R.”</span> Edited by Russell Schwartz. <em>Bioinformatics</em> 35 (3): 526–28. <a href="https://doi.org/10.1093/bioinformatics/bty633">https://doi.org/10.1093/bioinformatics/bty633</a>.
</div>
<div id="ref-yu2016" class="csl-entry">
Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2016. <span>“Ggtree: An r Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.”</span> Edited by Greg McInerny. <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210x.12628">https://doi.org/10.1111/2041-210x.12628</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sec-data-conversion.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sec-genetic-diversity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-PopulationStructure.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
