<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Get Started üöÄ | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="There are two easy ways to install and run ShiNyP: Using R: This method is suitable if you already have R installed or prefer working within the R environment. You‚Äôll need to install some R...">
<meta name="generator" content="bookdown 0.42 with bs4_book()">
<meta property="og:title" content="Get Started üöÄ | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/get-started.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="There are two easy ways to install and run ShiNyP: Using R: This method is suitable if you already have R installed or prefer working within the R environment. You‚Äôll need to install some R...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Get Started üöÄ | User Guide for ShiNyP">
<meta name="twitter:description" content="There are two easy ways to install and run ShiNyP: Using R: This method is suitable if you already have R installed or prefer working within the R environment. You‚Äôll need to install some R...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Noto_Sans-0.4.9/font.css" rel="stylesheet">
<link href="libs/Ubuntu_Sans-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="active" href="get-started.html">Get Started üöÄ</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="" href="sec-index.html"><span class="header-section-number">9</span> FAQ</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="get-started" class="section level1 unnumbered">
<h1>
<strong>Get Started</strong> üöÄ<a class="anchor" aria-label="anchor" href="#get-started"><i class="fas fa-link"></i></a>
</h1>
<p>There are two easy ways to install and run <em>ShiNyP</em>:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Using R</strong>:<br>
This method is suitable if you already have R installed or prefer
working within the R environment. You‚Äôll need to install some R
packages and then launch <em>ShiNyP</em> directly from R environment. ‚ñ∏
R/RStudio/Rtools Tutorial for Beginners!</p></li>
<li><p><strong>Using Docker</strong>:<br>
This is the simpler option if you‚Äôd rather skip installing R or any
packages. With Docker, you can run <em>ShiNyP</em> in a ready-to-use setup
with just one command. ‚ñ∏ Docker Tutorial for Beginners!</p></li>
</ol>
<p><br></p>
<hr>
<div id="sec-run-shinyp-via-r" class="section level2 unnumbered">
<h2>üîò Run <em>ShiNyP</em> via R<a class="anchor" aria-label="anchor" href="#sec-run-shinyp-via-r"><i class="fas fa-link"></i></a>
</h2>
<div id="prerequisites" class="section level4 unnumbered">
<h4>
<strong>‚úÖ Prerequisites</strong><a class="anchor" aria-label="anchor" href="#prerequisites"><i class="fas fa-link"></i></a>
</h4>
<p>Before installing <em>ShiNyP</em>, ensure your system meets the following
requirements:</p>
<ul>
<li>
<p><a href="https://www.r-project.org/"><strong>R</strong></a>: Version ‚â• 4.4.</p>
<p>Check your current version in R: <strong><code><a href="https://rdrr.io/r/base/numeric_version.html">getRversion()</a></code></strong></p>
</li>
<li>
<p><a href="https://www.bioconductor.org/install/"><strong>Bioconductor</strong></a>: Version ‚â•
3.20.</p>
<p>Match your Bioconductor version with your R version (e.g., use
Bioconductor 3.21 if R = 4.5).</p>
</li>
</ul>
</div>
<div id="install-required-package" class="section level4 unnumbered">
<h4>
<strong>1Ô∏è‚É£ Install Required Package</strong><a class="anchor" aria-label="anchor" href="#install-required-package"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html">install</a></span><span class="op">(</span>version <span class="op">=</span> <span class="st">"3.21"</span><span class="op">)</span> <span class="co"># Use the version that matches your R</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"qvalue"</span>, <span class="st">"SNPRelate"</span>, <span class="st">"ggtree"</span>, <span class="st">"snpStats"</span><span class="op">)</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="install-the-shinyp-package" class="section level4 unnumbered">
<h4>
<strong>2Ô∏è‚É£ Install the <em>ShiNyP</em> Package</strong><a class="anchor" aria-label="anchor" href="#install-the-shinyp-package"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"TeddYenn/ShiNyP"</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="start-the-shinyp-platform" class="section level4 unnumbered">
<h4>
<strong>3Ô∏è‚É£ Start the <em>ShiNyP</em> Platform</strong><a class="anchor" aria-label="anchor" href="#start-the-shinyp-platform"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ShiNyP</span><span class="op">)</span></span>
<span><span class="fu">ShiNyP</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ShiNyP/man/run_ShiNyP.html">run_ShiNyP</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div id="run-analysis-on-shinyp" class="section level4 unnumbered">
<h4>
<strong>4Ô∏è‚É£ Run Analysis on <em>ShiNyP</em></strong><a class="anchor" aria-label="anchor" href="#run-analysis-on-shinyp"><i class="fas fa-link"></i></a>
</h4>
<p>Input your SNP dataset in VCF, or try the built-in <span style="color: #1e90ff;"><strong>Demo
Data</strong></span>.</p>
<div class="inline-figure"><img src="images/1.2-R.gif" style="width:6.8in"></div>
<hr>
<p><br></p>
</div>
</div>
<div id="sec-run-shinyp-via-docker" class="section level2 unnumbered">
<h2>üîò Run <em>ShiNyP</em> via Docker<a class="anchor" aria-label="anchor" href="#sec-run-shinyp-via-docker"><i class="fas fa-link"></i></a>
</h2>
<p>If you have üê≥ Docker installed, you can launch¬†<em>ShiNyP</em>¬†without
installing R.</p>
<div id="prerequisite" class="section level4 unnumbered">
<h4>
<strong>‚úÖ Prerequisite</strong><a class="anchor" aria-label="anchor" href="#prerequisite"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<p><a href="https://www.docker.com/">Docker</a>.</p>
<p>Verify your Docker installation in Terminal: <strong><code>docker --version</code></strong></p>
</li>
</ul>
</div>
<div id="pull-the-docker-image" class="section level4 unnumbered">
<h4>
<strong>1Ô∏è‚É£ Pull the Docker Image</strong><a class="anchor" aria-label="anchor" href="#pull-the-docker-image"><i class="fas fa-link"></i></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="get-started.html#cb4-1" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-d</span> <span class="at">-p</span> 3838:3838 teddyenn/shinyp-platform</span></code></pre></div>
</div>
<div id="start-the-shinyp-platform-1" class="section level4 unnumbered">
<h4>
<strong>2Ô∏è‚É£ Start the <em>ShiNyP</em> Platform</strong><a class="anchor" aria-label="anchor" href="#start-the-shinyp-platform-1"><i class="fas fa-link"></i></a>
</h4>
<p>Open your browser and visit üëâ
<a href="http://localhost:3838/">http://localhost:3838</a></p>
<div class="inline-figure"><img src="images/1.3-Docker.gif" style="width:6.8in"></div>
<hr>
<p><br></p>
</div>
</div>
<div id="main-features" class="section level2 unnumbered">
<h2>Main Features<a class="anchor" aria-label="anchor" href="#main-features"><i class="fas fa-link"></i></a>
</h2>
<div class="inline-figure"><img src="images/Graphical%20Abstract.jpg" width="680"></div>
<div class="line-block">üîº <strong>Overview of the <em>ShiNyP</em> Platform Workflow for SNP Analysis.</strong><br>
‚ñ∏ <strong>Data Input &amp; Processing:</strong><br>
The workflow begins with Variant Call Format (VCF) <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html">Data Input</a>, followed by essential steps such as <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html">Data Quality Control (QC)</a> and <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html">Data Transformation</a> to prepare the data for analysis.<br>
‚ñ∏ <strong>Modular Analysis &amp; Output:</strong><br>
Analytical functions are organized into distinct modules‚Äîeach accessible as a separate page within the platform. These include: <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html">Population Structure</a>, <a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html">Genetic Diversity</a>, <a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html">Selection Sweep</a>, and <a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html">Core Collection</a>. Each module contains multiple subpages offering specialized tools for detailed analysis.<br>
‚ñ∏ <strong>Customizable Output:</strong><br><em>ShiNyP</em> delivers publication-ready visualizations and AI-generated reports that summarize analytical results in clear, structured narratives. Users can tailor output formats to fit specific research needs, ensuring both flexibility and interpretability.</div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html">Welcome to ShiNyP</a></div>
<div class="next"><a href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#get-started">Get Started üöÄ</a></li>
<li><a class="nav-link" href="#sec-run-shinyp-via-r">üîò Run ShiNyP via R</a></li>
<li><a class="nav-link" href="#sec-run-shinyp-via-docker">üîò Run ShiNyP via Docker</a></li>
<li><a class="nav-link" href="#main-features">Main Features</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/index.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/index.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on Apr 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
