<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>1 Data Input | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="➡️ This section contains two subpages: VCF and data.frame/genlight, allowing you to upload various data types for analysis.   1.1 VCF  Required Dataset (one of the following): VCF file from PLINK...">
<meta name="generator" content="bookdown 0.42 with bs4_book()">
<meta property="og:title" content="1 Data Input | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-data-input.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="➡️ This section contains two subpages: VCF and data.frame/genlight, allowing you to upload various data types for analysis.   1.1 VCF  Required Dataset (one of the following): VCF file from PLINK...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="1 Data Input | User Guide for ShiNyP">
<meta name="twitter:description" content="➡️ This section contains two subpages: VCF and data.frame/genlight, allowing you to upload various data types for analysis.   1.1 VCF  Required Dataset (one of the following): VCF file from PLINK...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Lato-0.4.9/font.css" rel="stylesheet">
<link href="libs/Roboto_Mono-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="" href="sec-quickstart.html">Quickstart</a></li>
<li><a class="active" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="" href="sec-index.html"><span class="header-section-number">9</span> INDEX</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-data-input" class="section level1" number="1">
<h1>
<span class="header-section-number">1</span> Data Input<a class="anchor" aria-label="anchor" href="#sec-data-input"><i class="fas fa-link"></i></a>
</h1>
<div style="background-color: #7c6f41; color: white; padding: 10px 5px; margin: 10px 0px; border-radius: 15px;">
<p>➡️ This section contains two subpages: <u><strong>VCF</strong></u> and
<u><strong>data.frame/genlight</strong></u>, allowing you to upload various
data types for analysis.</p>
</div>
<div class="inline-figure"><img src="images/Supplementary%20Fig.%201.jpg" width="700"></div>
<hr>
<div id="vcf" class="section level2" number="1.1">
<h2>
<span class="header-section-number">1.1</span> VCF<a class="anchor" aria-label="anchor" href="#vcf"><i class="fas fa-link"></i></a>
</h2>
<div id="required-dataset-one-of-the-following" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset (one of the following):<a class="anchor" aria-label="anchor" href="#required-dataset-one-of-the-following"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p>VCF file from
<a href="https://www.cog-genomics.org/plink/1.9/input" target="_blank">PLINK</a></p></li>
<li><p>VCF or gzipped VCF (vcf.gz) file from
<a href="https://vcftools.sourceforge.net/man_latest.html" target="_blank">VCFtools</a></p></li>
<li><p>VCF file in RDS format from <span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span></p></li>
</ul>
<p>The VCF file should contain chromosome and position information in the
first two columns (<span style="color: #8b0000;"><strong><code>#CHROM</code></strong></span> and
<span style="color: #8b0000;"><strong><code>POS</code></strong></span>), along with sample names and their
genotypic information. For some whole genome sequencing (WGS) data,
where SNP marker ID information is missing,
<span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span> will auto-generate the SNP ID
names as <u>#CHROM:POS</u>, such as <u>2:12500</u>,
indicating chromosome 2, position 12500.</p>
<details><summary>
⚙️ <strong>Using Beagle for Imputation</strong>
</summary><p><a href="https://faculty.washington.edu/browning/beagle/beagle_5.4_18Mar22.pdf"><strong>Beagle</strong></a>
is a widely used tool for phasing <span class="citation">(<a href="sec-index.html#ref-browning2021">Browning et al. 2021</a>)</span>, genotype imputation
<span class="citation">(<a href="sec-index.html#ref-browning2018">Browning, Zhou, and Browning 2018</a>)</span>, and haplotype inference in diploid and polyploid
species. It supports VCF input/output and uses a Hidden Markov Model
(HMM) for imputation.</p>
<p><strong>Step 1: Installation</strong></p>
<p>Download the latest Beagle JAR file:
<a href="https://faculty.washington.edu/browning/beagle/beagle.html" class="uri">https://faculty.washington.edu/browning/beagle/beagle.html</a></p>
<p><strong>Step 2:</strong> <strong>Input Files</strong></p>
<ol style="list-style-type: decimal">
<li><p>Genotype file (<code>gt=your_data.vcf.gz</code>): Contains the target genotypes
to be imputed.</p></li>
<li><p>Reference panel (<code>ref=reference.vcf.gz</code>): A phased reference panel
(optional).</p></li>
<li><p>Genetic map (<code>map=genetic_map.txt</code>): PLINK format genetic map
(optional).</p></li>
</ol>
<p>Compress and index input files:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="sec-data-input.html#cb15-1" tabindex="-1"></a><span class="ex">bgzip</span> your_data.vcf</span>
<span id="cb15-2"><a href="sec-data-input.html#cb15-2" tabindex="-1"></a><span class="ex">tabix</span> <span class="at">-p</span> vcf your_data.vcf.gz</span></code></pre></div>
<p><strong>Step 3:</strong> <strong>Running Beagle for Imputation</strong></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb16-1"><a href="sec-data-input.html#cb16-1" tabindex="-1"></a><span class="ex">java</span> <span class="at">-Xmx8g</span> <span class="at">-jar</span> beagle.5.4.jar <span class="dt">\</span></span>
<span id="cb16-2"><a href="sec-data-input.html#cb16-2" tabindex="-1"></a>  gt=your_data.vcf.gz <span class="dt">\</span></span>
<span id="cb16-3"><a href="sec-data-input.html#cb16-3" tabindex="-1"></a>  ref=reference.vcf.gz <span class="dt">\</span></span>
<span id="cb16-4"><a href="sec-data-input.html#cb16-4" tabindex="-1"></a>  map=genetic_map.txt <span class="dt">\</span></span>
<span id="cb16-5"><a href="sec-data-input.html#cb16-5" tabindex="-1"></a>  out=imputed_output <span class="dt">\</span></span>
<span id="cb16-6"><a href="sec-data-input.html#cb16-6" tabindex="-1"></a>  nthreads=8 <span class="dt">\</span></span>
<span id="cb16-7"><a href="sec-data-input.html#cb16-7" tabindex="-1"></a>  iterations=20 <span class="dt">\</span></span>
<span id="cb16-8"><a href="sec-data-input.html#cb16-8" tabindex="-1"></a>  window=50</span></code></pre></div>
<p><code>gt=your_data.vcf.gz</code> → Input genotype file (your VCF dataset).<br><code>ref=reference.vcf.gz</code> → Reference panel (must be phased and non-missing
alleles).<br><code>map=genetic_map.txt</code> → Genetic map (If no map is specified, Beagle
assumes a constant recombination rate of 1 cM per Mb).</p>
<p><code>nthreads=8</code> → Genetic map (uses more CPU threads for faster
computation.).<br><code>window=50</code> → Sets window size (default = 40 cM).<br><code>iterations=20</code> → Increases iterations for better accuracy (default =
12).</p>
<p><code>out=imputed_output</code> → Output file prefix.</p>
<p><strong>Output Files</strong></p>
<ul>
<li><p><code>imputed_output.vcf.gz</code>: Imputed genotype VCF file.</p></li>
<li><p><code>imputed_output.log</code>: Log file with summary statistics.</p></li>
</ul></details><hr>
</div>
<div id="step-1-input-your-vcf-file" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 1: Input your VCF File<a class="anchor" aria-label="anchor" href="#step-1-input-your-vcf-file"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #e7eaed;">Browse</span> and upload one VCF
file.</p></li>
<li><p>If your VCF file is from VCFtools, please tick the ‘VCF file from
VCFtools’ checkbox.</p></li>
<li><p>After the progress bar shows ‘Upload complete’, click the <span style="color: #fff;background-color: #007ACC;"><strong>Input
VCF File</strong></span> button.</p></li>
</ol>
<p><em>Or use our Demo Data</em></p>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Use Demo
Data</strong></span> button and
select one species. Detailed descriptions of the demo datasets are
available at
<a href="https://github.com/TeddYenn/ShiNyP/tree/main/inst/demo_data" class="uri">https://github.com/TeddYenn/ShiNyP/tree/main/inst/demo_data</a>.</li>
</ol>
<blockquote>
<p><strong>Note:</strong> By default, the genotypic information for 5 samples and 10
SNPs will be displayed on the interactive table.</p>
</blockquote>
<hr>
</div>
<div id="step-2-transform-to-data.frame" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 2: Transform to data.frame<a class="anchor" aria-label="anchor" href="#step-2-transform-to-data.frame"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p>If you have already input a VCF file on
<span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span>, click the <span style="color: #fff;background-color: #007ACC;"><strong>Transform to
data.frame</strong></span>
button.</p></li>
<li><p>Download the <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> file (in
RDS format) and Site Info (in RDS format) so that you will not have
to input the VCF file again; instead, you can upload the
<span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> file.</p></li>
</ol>
<blockquote>
<p><strong>Note:</strong> For the current version (v0.1.1),
<span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span> is optimized for <strong>genome-wide
SNP</strong> analysis in <strong>diploid species</strong>, supporting a wide range of
diploid taxa, including plants, animals, and microbial eukaryotes. If
you have polyploid or haploid dataset, please tick the ‘Applying
diploidization processing’ checkbox.</p>
</blockquote>
<hr>
</div>
<div id="outputs" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>VCF Data (RDS)</strong>: VCF data stored in RDS format, which can be open
and read in R environment.</p></li>
<li><p><strong>data.frame (RDS)</strong>: <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span>
file. It’s necessary for downstream analyses, <em>please download and
save it!</em></p></li>
<li><p><strong>Site Info. (RDS)</strong>: SNP site information file. It’s necessary for
downstream analyses, <em>please download and save it!</em></p></li>
</ul>
<blockquote>
<p><strong>Note:</strong> If your data is large (more than 1GB), it may take some time
to process. Please be patient. The
<span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span> platform processes one task at
a time (e.g., you must wait for the input process to finish before you
can reset the data).</p>
</blockquote>
<div class="inline-figure"><img src="screenshots/DataInput1.png" width="650"></div>
<p><em>VCF Data Input!</em></p>
<hr>
</div>
</div>
<div id="data.framegenlight" class="section level2" number="1.2">
<h2>
<span class="header-section-number">1.2</span> data.frame/genlight<a class="anchor" aria-label="anchor" href="#data.framegenlight"><i class="fas fa-link"></i></a>
</h2>
<div id="required-dataset" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#required-dataset"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> in RDS file format</p></li>
<li><p><span style="color: #8b0000;"><strong><code>genlight</code></strong></span> in RDS file format</p></li>
</ul>
<div class="line-block">
<span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> file can be downloaded from the subpages <u>VCF</u>, <u>Sample QC</u>, <u>SNP QC</u>, <u>Core Sample Set,</u> and <u>Core SNP Set</u>.<br><span style="color: #8b0000;"><strong><code>genlight</code></strong></span> files can be downloaded from the <u>Data Transform</u> page.</div>
<hr>
</div>
<div id="one-step" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>One Step:</strong><a class="anchor" aria-label="anchor" href="#one-step"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li>
<span style="background-color: #e7eaed;">Browse</span> and click the
<span style="color: #fff;background-color: #007ACC;"><strong>Input</strong></span> button
to upload your <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> and
<span style="color: #8b0000;"><strong><code>genlight</code></strong></span> files.</li>
</ol>
<div class="inline-figure"><img src="screenshots/DataInput2.png" width="650"></div>
<p><span style="color: #8b0000;"><strong><em><code>data.frame</code></em></strong></span><em>/<span style="color: #8b0000;"><strong><code>genlight</code></strong></span>
Data Input!</em></p>
<hr>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-quickstart.html">Quickstart</a></div>
<div class="next"><a href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-data-input"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="nav-link" href="#vcf"><span class="header-section-number">1.1</span> VCF</a></li>
<li><a class="nav-link" href="#data.framegenlight"><span class="header-section-number">1.2</span> data.frame/genlight</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/02-DataInput.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/02-DataInput.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on Feb 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
