<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>4 Population Structure | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="➡️ This section contains seven subpages: PCA, DAPC, UPGMA Tree, NJ Tree, Kinship, Scatter PlotPlus, and Tree PlotPlus allowing you to conduct various population structure analyses and customize...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="4 Population Structure | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-population-structure.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="➡️ This section contains seven subpages: PCA, DAPC, UPGMA Tree, NJ Tree, Kinship, Scatter PlotPlus, and Tree PlotPlus allowing you to conduct various population structure analyses and customize...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="4 Population Structure | User Guide for ShiNyP">
<meta name="twitter:description" content="➡️ This section contains seven subpages: PCA, DAPC, UPGMA Tree, NJ Tree, Kinship, Scatter PlotPlus, and Tree PlotPlus allowing you to conduct various population structure analyses and customize...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Lato-0.4.9/font.css" rel="stylesheet">
<link href="libs/Roboto_Mono-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="" href="sec-quickstart.html">Quickstart</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="active" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="" href="sec-index.html"><span class="header-section-number">9</span> INDEX</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-population-structure" class="section level1" number="4">
<h1>
<span class="header-section-number">4</span> Population Structure<a class="anchor" aria-label="anchor" href="#sec-population-structure"><i class="fas fa-link"></i></a>
</h1>
<div class="highlighted-text" style="background-color: #7c6f41; color: white; padding: 10px 5px; margin: 10px 0px; border-radius: 15px;">
<p>➡️ This section contains seven subpages: <u><strong>PCA</strong></u>,
<u><strong>DAPC</strong></u>, <u><strong>UPGMA Tree</strong></u>, <u><strong>NJ
Tree</strong></u>, <u><strong>Kinship</strong></u>, <u><strong>Scatter
Plot</strong></u><sup><strong>Plus</strong></sup>, and <u><strong>Tree Plot</strong></u><sup><strong>Plus</strong></sup>
allowing you to conduct various population structure analyses and
customize your plot.</p>
</div>
<div class="inline-figure"><img src="images/Supplementary%20Fig.%202.jpg" width="700"></div>
<div id="pca-principal-component-analysis" class="section level2" number="4.1">
<h2>
<span class="header-section-number">4.1</span> PCA (Principal Component Analysis)<a class="anchor" aria-label="anchor" href="#pca-principal-component-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>A widely used method to uncover underlying population structure by
reducing the dimensionality of genetic data.</p>
<div id="required-dataset-1" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#required-dataset-1"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
</ul>
<hr>
</div>
<div id="one-step-1" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>One Step:</strong><a class="anchor" aria-label="anchor" href="#one-step-1"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run
PCA</strong></span> button to
generate PCA plots and the following downloadable files.</li>
</ol>
<blockquote>
<p><strong>Note</strong>: You can upload the <strong>Group Info.</strong> (from <u>Population
Structure</u>/<u>DAPC</u>) or <strong>Core Sample Info.</strong>
(from <u>Core Collection</u>/<u>Core Sample Set</u>) to
classify individuals and color them in the PCA Scatter Plot.</p>
</blockquote>
<hr>
</div>
<div id="outputs-5" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-5"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>PCA Scatter Plot (PDF)</strong>: A scatter plot showing the distribution
of samples based on principal components, with each dot representing
an individual.</p></li>
<li><p><strong>PC Explained Variance Plot (PDF)</strong>: Visualizes the variance
explained by each principal component.</p></li>
<li><p><strong>Explained Variance (CSV)</strong>: Contains the explained variance of
each principal component.</p></li>
<li><p><strong>PCA Transformed Data (CSV)</strong>: Dataset transformed into principal
components, with samples as rows and principal components as
columns.</p></li>
<li><p><strong>PCA Object (RDS)</strong>: Contains all PCA results for future use and
reproducibility, and can be used as input data in the <u>Population
Structure</u>/<u>Scatter Plot</u><sup>Plus</sup> subpage.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/PopulationStructure1.png" width="650"></div>
<p><em>PCA Complete!</em></p>
<hr>
</div>
</div>
<div id="dapc-discriminant-analysis-of-principal-components" class="section level2" number="4.2">
<h2>
<span class="header-section-number">4.2</span> DAPC (Discriminant Analysis of Principal Components)<a class="anchor" aria-label="anchor" href="#dapc-discriminant-analysis-of-principal-components"><i class="fas fa-link"></i></a>
</h2>
<p>A multivariate method for identifying and visualizing genetic clusters
by combining PCA and Linear Discriminant Analysis (LDA) <span class="citation">(<a href="sec-index.html#ref-jombart2010">Jombart, Devillard, and Balloux 2010</a>)</span>.
For more information, visit
<a href="https://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf" target="_blank">https://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf</a>.</p>
<div id="required-dataset-2" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#required-dataset-2"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><span style="color: #8b0000;"><strong><code>genlight</code></strong></span></li>
<li><strong><code>genind</code></strong></li>
</ul>
<hr>
</div>
<div id="step-1-cluster-identification" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 1: <strong>Cluster Identification</strong><a class="anchor" aria-label="anchor" href="#step-1-cluster-identification"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run DAPC
I</strong></span> button to
determine the optimal number of clusters (the lowest BIC value
indicates the optimal number of clusters).</li>
</ol>
<blockquote>
<p><strong>Note:</strong> The default number of PC axes for cluster identification is
set to retain PCs that capture up to 80% of the total variance. You
can refer the “PC Explained Variance Plot” in the <u>Population
Structure</u>/<u>PCA</u> subpage.</p>
</blockquote>
<hr>
</div>
<div id="step-2-dapc-analysis" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 2: <strong>DAPC Analysis</strong><a class="anchor" aria-label="anchor" href="#step-2-dapc-analysis"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p>Choose the number of cluster (K) based on the “Bayesian Information
Criterion (BIC) Plot”.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run DAPC
II</strong></span> button to
generate DAPC plots and the following downloadable files.</p></li>
</ol>
<blockquote>
<p><strong>Note</strong>: You can download the “DAPC Object” and upload it on
<u>Population Structure</u>/<u>Scatter Plot</u><sup>Plus</sup>
subpage to customize your 2D and 3D scatter plots.</p>
</blockquote>
<hr>
</div>
<div id="outputs-6" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-6"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>Bayesian Information Criterion (BIC) Plot (PDF)</strong>: Visual
representation of the BIC for model selection.</p></li>
<li><p><strong>Density Plot of First &amp; Second Discriminant Function</strong> <strong>(PDF)</strong>:
Displays the density of the first and second discriminant functions,
with each row bar representing an individual.</p></li>
<li><p><strong>DAPC Scatter Plot (PDF)</strong>: A scatter plot showing the distribution
of samples based on discriminant functions (x-axis: first
discriminant function; y-axis: second discriminant function), with
each dot representing an individual.</p></li>
<li><p><strong>DAPC Membership Probability Plot (PDF)</strong>: Visualizes membership
probabilities of individuals in different groups, with each row bar
representing an individual.</p></li>
<li><p><strong>DAPC Group Info. (CSV)</strong>: Contains the group assignments for each
individual based on DAPC. This file used in various subpages.</p></li>
<li><p><strong>DAPC Transformed Data (CSV)</strong>: Dataset transformed into
discriminant functions with samples as rows and discriminant
functions as columns.</p></li>
<li><p><strong>DAPC Object (RDS)</strong>: Contains all results from the DAPC analysis
for future reproducibility. It can be used as input data in the
<u>Population Structure</u>/<u>Scatter Plot</u><sup>Plus</sup>
and <u>Core Collection</u>/<u>Core SNP Set</u>
subpages.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/PopulationStructure2.png" width="650"></div>
<p><em>DAPC Complete!</em></p>
<hr>
</div>
</div>
<div id="upgma-unweighted-pair-group-method-with-arithmetic-mean-tree" class="section level2" number="4.3">
<h2>
<span class="header-section-number">4.3</span> UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree<a class="anchor" aria-label="anchor" href="#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree"><i class="fas fa-link"></i></a>
</h2>
<p>A classic approaach for constructing rooted trees based on genetic
distance data. UPGMA tree is generated by <em>poppr</em> and <em>ggtree</em> packages
<span class="citation">(<a href="sec-index.html#ref-yu2016">Yu et al. 2016</a>; <a href="sec-index.html#ref-kamvar2014">Kamvar, Tabima, and Grünwald 2014</a>)</span>.</p>
<div id="required-dataset-3" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#required-dataset-3"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><span style="color: #8b0000;"><strong><code>genlight</code></strong></span></li>
</ul>
<hr>
</div>
<div id="steps-1" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps</strong>:<a class="anchor" aria-label="anchor" href="#steps-1"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p>Choose the number of bootstrap replicates, which will be used for
assessing the confidence of the branching structure.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run
UPGMA</strong></span> button to
generate tree plot.</p></li>
</ol>
<blockquote>
<p><strong>Note</strong>: You can download the “UPGMA Object” and upload it on
<u>Population Structure</u>/<u>Tree Plot</u><sup>Plus</sup>
subpage to customize your phylogenetic tree.</p>
</blockquote>
<hr>
</div>
<div id="outputs-7" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-7"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>UPGMA Phylogenetic Tree (PDF)</strong>: A UPGMA rooted tree with a
user-defined layout style.</p></li>
<li><p><strong>UPGMA Object (RDS)</strong>: Contains all information of the UPGMA tree,
and can be used as input data in the <u>Population
Structure</u>/<u>Tree Plot</u><sup>Plus</sup> subpage.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/PopulationStructure3.png" width="650"></div>
<p><em>UPGMA Tree Complete!</em></p>
<hr>
</div>
</div>
<div id="nj-neighbor-joining-tree" class="section level2" number="4.4">
<h2>
<span class="header-section-number">4.4</span> NJ (Neighbor-Joining) Tree<a class="anchor" aria-label="anchor" href="#nj-neighbor-joining-tree"><i class="fas fa-link"></i></a>
</h2>
<p>A method for building unrooted trees using genetic distance data. NJ
tree is generated by <em>ape</em> and <em>ggtree</em> packages <span class="citation">(<a href="sec-index.html#ref-paradis2018">Paradis and Schliep 2018</a>; <a href="sec-index.html#ref-yu2016">Yu et al. 2016</a>)</span>.</p>
<div id="required-dataset-4" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#required-dataset-4"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><span style="color: #8b0000;"><strong><code>genlight</code></strong></span></li>
</ul>
<hr>
</div>
<div id="one-step-2" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>One Step:</strong><a class="anchor" aria-label="anchor" href="#one-step-2"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run
NJ</strong></span> button to
generate tree plot.</li>
</ol>
<blockquote>
<p><strong>Note</strong>: You can download the “NJ Object” and upload it on
<u>Population Structure</u>/<u>Tree Plot</u><sup>Plus</sup>
subpage to customize your phylogenetic tree.</p>
</blockquote>
<hr>
</div>
<div id="outputs-8" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-8"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>NJ Phylogenetic Tree (PDF)</strong>: A NJ unrooted tree with a
user-defined layout style.</p></li>
<li><p><strong>NJ Object (RDS)</strong>: Contains all information of the NJ tree, and
can be used as input data in the <u>Population
Structure</u>/<u>Tree Plot</u><sup>Plus</sup> subpage.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/PopulationStructure4.png" width="650"></div>
<p><em>NJ Tree Complete!</em></p>
<hr>
</div>
</div>
<div id="kinship-analysis" class="section level2" number="4.5">
<h2>
<span class="header-section-number">4.5</span> Kinship Analysis<a class="anchor" aria-label="anchor" href="#kinship-analysis"><i class="fas fa-link"></i></a>
</h2>
<p>A statistical method for assessing genetic relationships and relatedness
among individuals based on shared alleles <span class="citation">(<a href="sec-index.html#ref-kang2010">Kang et al. 2010</a>)</span>. Kinship matrix is
generated by <em>statgenGWAS</em> package.For more information, visit
<a href="https://rdrr.io/cran/statgenGWAS/man/kinship.html" target="_blank">https://rdrr.io/cran/statgenGWAS/man/kinship.html</a>.</p>
<div id="required-dataset-5" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#required-dataset-5"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
</ul>
<hr>
</div>
<div id="steps-2" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-2"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Group Info.</strong> from
<u>Population Structure</u>/<u>DAPC</u> subpage
(optional). If uploaded, the order of samples will follow the group
assignment; otherwise, it will follow the order of the original VCF
data.</p></li>
<li><p>Choose a method to run kinship analysis.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run
Kinship</strong></span> button to
generate the kinship matrix.</p></li>
</ol>
<hr>
</div>
<div id="outputs-9" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-9"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>Kinship Matrix Plot (PDF)</strong>: A visual representation of the
kinship matrix.</p></li>
<li><p><strong>Kinship Matrix (RDS)</strong>: Contains the kinship matrix data.</p></li>
</ul>
<blockquote>
<p>Note: This kinship matrix can be directly used as input for <em>GAPIT</em>
package in genome-wide association studies (GWAS), helping to control
for confounding effects.</p>
</blockquote>
<div class="inline-figure"><img src="screenshots/PopulationStructure5.png" width="650"></div>
<p><em>Kinship Analysis Complete!</em></p>
<hr>
</div>
</div>
<div id="scatter-plot-plus" class="section level2" number="4.6">
<h2>
<span class="header-section-number">4.6</span> Scatter Plot <sup>Plus</sup><a class="anchor" aria-label="anchor" href="#scatter-plot-plus"><i class="fas fa-link"></i></a>
</h2>
<p>Customize your scatter plot based on the results from <u>Population
Structure</u>/<u>PCA</u> or <u>Population
Structure</u>/<u>DAPC</u>.</p>
<div id="required-files" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Files:<a class="anchor" aria-label="anchor" href="#required-files"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<strong>PCA Object</strong> (PCA_prcomp_Object.rds file) or <strong>DAPC Object</strong>
(DAPC_dapc_Object.rds file)</li>
<li>
<strong>Group and Other Info.</strong> (i.e. metadata, modifiable from
DAPC_Group_Info.csv)</li>
</ul>
<blockquote>
<p><strong>Note:</strong> If you are using a <strong>DAPC object</strong> as input, ensure that
there are <strong>at least 4 groups</strong>. Since an N-group dataset has N-1 LD
axes, you need more than 3 axes to generate a 3D scatter plot.</p>
</blockquote>
<blockquote>
<p><strong>Note</strong>: You can add metadata (more information about samples) by
adding new variables to the Group Info. file. Ensure that the sample
order remains unchanged.</p>
</blockquote>
<details><summary>
➡️ <strong>Example: Group Info. file (CSV)</strong>
</summary><div class="inline-figure"><img src="images/Chromosome-Info.png" width="350"></div>
</details><hr>
</div>
<div id="steps-3" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-3"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>PCA or DAPC Object
(RDS)</strong></p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Group and Other Info.
(CSV)</strong></p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Scatter
Plot</strong></span> button to
generate the 2D and 3D interactive scatter plots.</p></li>
<li><p>Customize the scatter plot and click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Scatter
Plot</strong></span> button
again.</p></li>
</ol>
<blockquote>
<p><strong>Note</strong>: The scatter plots are downloaded as HTML files and can be
opened with browsers like Chrome or Edge.</p>
</blockquote>
<hr>
</div>
<div id="outputs-10" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-10"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>2D Scatter Plot (HTML)</strong>: Two-dimensional interactive scatter plot
with user-defined attributes.</p></li>
<li><p><strong>3D Scatter Plot (HTML)</strong>: Three-dimensional interactive scatter
plot with user-defined attributes.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/PopulationStructure6.png" width="650"></div>
<p><em>Scatter Plot <sup>Plus</sup> Complete!</em></p>
</div>
</div>
<div id="tree-plot-plus" class="section level2" number="4.7">
<h2>
<span class="header-section-number">4.7</span> Tree Plot <sup>Plus</sup><a class="anchor" aria-label="anchor" href="#tree-plot-plus"><i class="fas fa-link"></i></a>
</h2>
<p>Customize your phylogenetic tree plot based on the results from
<u>Population Structure</u>/<u>UPGMA</u> or <u>Population
Structure</u>/<u>NJ</u>.</p>
<div id="required-files-1" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Files:<a class="anchor" aria-label="anchor" href="#required-files-1"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<strong>UPGMA Object</strong> (UPGMA_phylo_Object.rds file) or <strong>NJ Object</strong>
(NJ_phylo_Object.rds file)</li>
<li>
<strong>Group and Other Info.</strong> (modifiable from DAPC_Group_Info.csv)</li>
</ul>
<blockquote>
<p><strong>Note</strong>: You can add more information about samples by adding new
variables to the Group Info. file. Ensure that the sample order
remains unchanged.</p>
</blockquote>
<hr>
</div>
<div id="steps-4" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-4"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>UPGMA or NJ Object
(RDS)</strong></p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Group and Other Info.
(CSV)</strong></p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Tree
Plot</strong></span> button to
generate the tree plot.</p></li>
<li><p>Customize the tree plot and click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Tree
Plot</strong></span> button
again.</p></li>
</ol>
<hr>
</div>
<div id="outputs-11" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-11"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<strong>Phylogenetic Tree Plot (PDF)</strong>: A phylogenetic tree plot with
user-defined layout style and attributes.</li>
</ul>
<div class="inline-figure"><img src="screenshots/PopulationStructure7.png" width="650"></div>
<p><em>Tree Plot <sup>Plus</sup> Complete!</em></p>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></div>
<div class="next"><a href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-population-structure"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="nav-link" href="#pca-principal-component-analysis"><span class="header-section-number">4.1</span> PCA (Principal Component Analysis)</a></li>
<li><a class="nav-link" href="#dapc-discriminant-analysis-of-principal-components"><span class="header-section-number">4.2</span> DAPC (Discriminant Analysis of Principal Components)</a></li>
<li><a class="nav-link" href="#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree"><span class="header-section-number">4.3</span> UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree</a></li>
<li><a class="nav-link" href="#nj-neighbor-joining-tree"><span class="header-section-number">4.4</span> NJ (Neighbor-Joining) Tree</a></li>
<li><a class="nav-link" href="#kinship-analysis"><span class="header-section-number">4.5</span> Kinship Analysis</a></li>
<li><a class="nav-link" href="#scatter-plot-plus"><span class="header-section-number">4.6</span> Scatter Plot Plus</a></li>
<li><a class="nav-link" href="#tree-plot-plus"><span class="header-section-number">4.7</span> Tree Plot Plus</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/05-PopulationStructure.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/05-PopulationStructure.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on Feb 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
