<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>2 Data QC | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); ‚û°Ô∏è This section includes three subpages: Sample...">
<meta name="generator" content="bookdown 0.43 with bs4_book()">
<meta property="og:title" content="2 Data QC | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-data-qc.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); ‚û°Ô∏è This section includes three subpages: Sample...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="2 Data QC | User Guide for ShiNyP">
<meta name="twitter:description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); ‚û°Ô∏è This section includes three subpages: Sample...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Noto_Sans-0.4.10/font.css" rel="stylesheet">
<link href="libs/Ubuntu_Sans-0.4.10/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.10/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link rel="shortcut icon" href="favicon/favicon.ico">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="" href="get-started.html">Get Started üöÄ</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="active" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="" href="sec-index.html"><span class="header-section-number">9</span> FAQ</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-data-qc" class="section level1" number="2">
<h1>
<span class="header-section-number">2</span> Data QC<a class="anchor" aria-label="anchor" href="#sec-data-qc"><i class="fas fa-link"></i></a>
</h1>
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-2662B66TL7"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2662B66TL7');
</script><div style="background-color: #7c6f41; color: white; padding: 10px 5px; margin: 10px 0px; border-radius: 15px;">
<div class="line-block">‚û°Ô∏è This section includes three subpages: <u>Sample QC</u>, <u>SNP QC</u>, and <u>SNP Density</u>, allowing you to assess the quality of samples and SNPs in <em>data.frame</em>, as well as visualize SNP density across the genome.</div>
</div>
<div id="sample-qc" class="section level2" number="2.1">
<h2>
<span class="header-section-number">2.1</span> Sample QC<a class="anchor" aria-label="anchor" href="#sample-qc"><i class="fas fa-link"></i></a>
</h2>
<div id="required-file-2" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required File:<a class="anchor" aria-label="anchor" href="#required-file-2"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<em>data.frame</em> file from the <u>Data Input</u> page or</li>
<li>SNP post-QC <em>data.frame</em> file from the <u>Data QC</u>/<u>SNP
QC</u> subpage .</li>
</ul>
<hr>
</div>
<div id="step-1-get-summary" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 1: Get Summary<a class="anchor" aria-label="anchor" href="#step-1-get-summary"><i class="fas fa-link"></i></a>
</h4>
<p>First, obtain the sample summary statistics (missing rate and
heterozygosity rate) by clicking both
<span style="color: #fff;background-color: #007ACC;">Summary</span> buttons and
you will see the results.</p>
<hr>
</div>
<div id="step-2-sample-qc" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 2: Sample QC<a class="anchor" aria-label="anchor" href="#step-2-sample-qc"><i class="fas fa-link"></i></a>
</h4>
<p>Adjust the thresholds and click <span style="color: #fff;background-color: #007ACC;">Sample QC by
Thresholds</span>. This will
generate the Post-QC <em>data.frame</em> file.</p>
<blockquote>
<p><strong>Note:</strong> If you prefer not to perform sample QC by sample missing
rate or heterozygosity rate, please set the threshold to 1.</p>
</blockquote>
<hr>
</div>
<div id="outputs-1" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-1"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<em>data.frame</em> (RDS): Updated <em>data.frame</em> file ‚Äî required for
downstream analysis.</li>
<li>Site Info. (RDS): Updated SNP site information file ‚Äî required for
downstream analysis.</li>
</ul>
<div class="inline-figure"><img src="images/3.1.gif" style="width:6.8in"></div>
<hr>
<p><br></p>
</div>
</div>
<div id="snp-qc" class="section level2" number="2.2">
<h2>
<span class="header-section-number">2.2</span> SNP QC<a class="anchor" aria-label="anchor" href="#snp-qc"><i class="fas fa-link"></i></a>
</h2>
<div id="required-file-3" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required File:<a class="anchor" aria-label="anchor" href="#required-file-3"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<em>data.frame</em> file from the <u>Data Input</u> page or</li>
<li>Sample post-QC <em>data.frame</em> file from the <u>Data
QC</u>/<u>Sample QC</u> subpage.</li>
</ul>
<hr>
</div>
<div id="step-1-get-summary-1" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 1: Get Summary<a class="anchor" aria-label="anchor" href="#step-1-get-summary-1"><i class="fas fa-link"></i></a>
</h4>
<p>First, obtain the SNP summary statistics [missing rate, minor allele
frequency (MAF), heterozygosity rate, and Hardy-Weinberg equilibrium
(HWE)] by clicking all
<span style="color: #fff;background-color: #007ACC;">Summary</span> buttons and
you will see the results.</p>
<hr>
</div>
<div id="step-2-sample-qc-1" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 2: Sample QC<a class="anchor" aria-label="anchor" href="#step-2-sample-qc-1"><i class="fas fa-link"></i></a>
</h4>
<p>Adjust the thresholds and click <span style="color: #fff;background-color: #007ACC;">SNP QC by
Thresholds</span>. This will
generate the Post-QC <em>data.frame</em> file.</p>
<blockquote>
<p><strong>Note:</strong> If you prefer not to perform QC based on SNP missing rate or
heterozygosity rate, set the missing rate threshold to 1, the MAF to
0, and the heterozygosity rate to 0 and 1. Also, leave the ‚ÄòDo SNP QC
by HWE‚Äô checkbox unticked to skip QC based on SNP HWE.</p>
</blockquote>
<hr>
</div>
<div id="outputs-2" class="section level4 unnumbered">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-2"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<em>data.frame</em> (RDS): Updated <em>data.frame</em> file ‚Äî required for
downstream analysis.</li>
<li>Site Info. (RDS): Updated SNP site information file ‚Äî required for
downstream analysis.</li>
</ul>
<div class="inline-figure"><img src="images/3.2.gif" style="width:6.8in"></div>
<hr>
</div>
</div>
<div id="snp-density" class="section level2" number="2.3">
<h2>
<span class="header-section-number">2.3</span> SNP Density<a class="anchor" aria-label="anchor" href="#snp-density"><i class="fas fa-link"></i></a>
</h2>
<div id="required-files" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Files:<a class="anchor" aria-label="anchor" href="#required-files"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p>Site Info. (RDS) of the current <em>data.frame</em>, downloadable from
<u>Data Input</u> or <u>Data QC</u> pages.</p></li>
<li>
<p>Chromosome Info. (CSV): Reference genome information of the current
study.</p>
<p><a href="https://github.com/TeddYenn/ShiNyP/blob/main/inst/demo_data/Wild_Rice/Chromosome_Info.csv"><strong><em>Click
here</em></strong></a><em>:
Download an example of Chromosome Info. (CSV).</em></p>
</li>
</ul>
<details><summary>
Example: Chromosome Info. of Rice
</summary><p>Data source:
<a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_034140825.1/" class="uri">https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_034140825.1/</a></p>
<table class="table table-sm">
<thead><tr class="header">
<th>Chr</th>
<th>Start</th>
<th>End</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Chr01</td>
<td>0</td>
<td>43929697</td>
</tr>
<tr class="even">
<td>Chr02</td>
<td>0</td>
<td>36447916</td>
</tr>
<tr class="odd">
<td>Chr03</td>
<td>0</td>
<td>37399924</td>
</tr>
<tr class="even">
<td>Chr04</td>
<td>0</td>
<td>36078568</td>
</tr>
<tr class="odd">
<td>Chr05</td>
<td>0</td>
<td>30400764</td>
</tr>
<tr class="even">
<td>Chr06</td>
<td>0</td>
<td>32122276</td>
</tr>
<tr class="odd">
<td>Chr07</td>
<td>0</td>
<td>29936421</td>
</tr>
<tr class="even">
<td>Chr08</td>
<td>0</td>
<td>28605474</td>
</tr>
<tr class="odd">
<td>Chr09</td>
<td>0</td>
<td>27474823</td>
</tr>
<tr class="even">
<td>Chr10</td>
<td>0</td>
<td>23931887</td>
</tr>
<tr class="odd">
<td>Chr11</td>
<td>0</td>
<td>31111469</td>
</tr>
<tr class="even">
<td>Chr12</td>
<td>0</td>
<td>28271460</td>
</tr>
</tbody>
</table></details><hr>
</div>
<div id="steps" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Steps:<a class="anchor" aria-label="anchor" href="#steps"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li>Upload Site Info. (RDS) and Chromosome Info. (CSV).</li>
<li>Choose a window size in kilobases (kb).</li>
<li>Click <span style="color: #fff;background-color: #007ACC;">Summary</span>.</li>
</ol>
<hr>
</div>
<div id="outputs-3" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-3"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>SNP Density Plot (PDF): An ideogram visualizing SNP density across
the genome within a defined window size. A gradient color palette is
used to represent varying SNP densities: <em>green</em> for <u>lower
densities</u>, <em>yellow</em> for <u>medium densities</u>,
and <em>red</em> for <u>higher densities</u>, with <em>grey</em> indicating
regions with <u>zero SNP</u>.</li>
<li>SNP Density (CSV): A table detailing SNP density across each
chromosome. <em>bp_over_SNPs</em>: The total base pairs (bp) per SNP in
each window, representing the average spacing between SNPs.
<em>SNPs_over_1000bp</em>: The number of SNPs per 1,000 base pairs,
providing a normalized measure of SNP density across the genome.</li>
</ul>
<div class="inline-figure"><img src="images/3.3.gif" style="width:6.8in;height:3.7in"></div>
<hr>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></div>
<div class="next"><a href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-data-qc"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="nav-link" href="#sample-qc"><span class="header-section-number">2.1</span> Sample QC</a></li>
<li><a class="nav-link" href="#snp-qc"><span class="header-section-number">2.2</span> SNP QC</a></li>
<li><a class="nav-link" href="#snp-density"><span class="header-section-number">2.3</span> SNP Density</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/03-DataQC.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/03-DataQC.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on May 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
