<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>6 Selection Sweep | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="➡️ This section contains four subpages: pcadapt, OutFLANK, IBS, and Manhattan PlotPlus, allowing you to detect selection signatures in different scenario and customize your plot.   6.1 pcadapt A...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="6 Selection Sweep | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-selection-sweep.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="➡️ This section contains four subpages: pcadapt, OutFLANK, IBS, and Manhattan PlotPlus, allowing you to detect selection signatures in different scenario and customize your plot.   6.1 pcadapt A...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="6 Selection Sweep | User Guide for ShiNyP">
<meta name="twitter:description" content="➡️ This section contains four subpages: pcadapt, OutFLANK, IBS, and Manhattan PlotPlus, allowing you to detect selection signatures in different scenario and customize your plot.   6.1 pcadapt A...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Lato-0.4.9/font.css" rel="stylesheet">
<link href="libs/Roboto_Mono-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="" href="sec-quickstart.html">Quickstart</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="active" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="" href="sec-index.html"><span class="header-section-number">9</span> INDEX</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-selection-sweep" class="section level1" number="6">
<h1>
<span class="header-section-number">6</span> Selection Sweep<a class="anchor" aria-label="anchor" href="#sec-selection-sweep"><i class="fas fa-link"></i></a>
</h1>
<div style="background-color: #7c6f41; color: white; padding: 10px 5px; margin: 10px 0px; border-radius: 15px;">
<p>➡️ This section contains four subpages: <u><strong>pcadapt</strong></u>,
<u><strong>OutFLANK</strong></u>, <u><strong>IBS</strong></u>, and <u><strong>Manhattan
Plot</strong></u><sup><strong>Plus</strong></sup>, allowing you to detect selection
signatures in different scenario and customize your plot.</p>
</div>
<div class="inline-figure"><img src="images/Supplementary%20Fig.%204.jpg" width="700"></div>
<div id="pcadapt" class="section level2" number="6.1">
<h2>
<span class="header-section-number">6.1</span> pcadapt<a class="anchor" aria-label="anchor" href="#pcadapt"><i class="fas fa-link"></i></a>
</h2>
<p>A PCA-based approach identifies selective outliers relative to
population structure <span class="citation">(<a href="sec-index.html#ref-Luu2017">Luu, Bazin, and Blum 2016</a>)</span>.</p>
<div id="required-datasets-2" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Datasets:<a class="anchor" aria-label="anchor" href="#required-datasets-2"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
<li>
<strong>Site Info.</strong> <strong>(RDS)</strong> of the current
<span style="color: #8b0000;"><strong><code>data.frame</code></strong></span>, downloadable from <u>Data
Input</u> or <u>Data QC</u> pages.</li>
</ul>
<hr>
</div>
<div id="steps-7" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-7"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Site Info.</strong>
(required).</p></li>
<li><p>Click <span style="color: #fff;background-color: #007ACC;">SNP Thinning</span>
button (optional) and choose window size (number of SNPs) and r²
threshold. For more information, visit
<a href="https://bcm-uga.github.io/pcadapt/articles/pcadapt.html" target="_blank">https://bcm-uga.github.io/pcadapt/articles/pcadapt.html</a>.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run
pcadapt</strong></span> button to
perform genome scan for selection.</p></li>
</ol>
<hr>
</div>
<div id="outputs-16" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-16"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>pcadapt p-value per site (RDS)</strong>: A dataset containing p-values
and adjusted p-values for each site.</p></li>
<li><p><strong>pcadapt Manhattan Plot (PDF)</strong>: A Manhattan plot visualizing the
p-values per site across the genome. Significant SNPs are
highlighted in red.</p></li>
<li><p><strong>pcadapt QQ Plot (PDF)</strong>: A QQ plot comparing the distribution of
observed p-values to the expected distribution under the null
hypothesis.</p></li>
<li><p><strong>pcadapt Histogram of p-values (PDF)</strong>: A histogram showing the
distribution of p-values across all sites.</p></li>
<li><p><strong>pcadapt Histogram of Test Statistics (PDF)</strong>: A histogram showing
the distribution of test statistics across all sites.</p></li>
<li><p><strong>pcadapt Significant SNPs (CSV)</strong>: A table listing SNPs identified
as significant by pcadapt, including their site info., p-values, and
adjusted p-values.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/SelectionSweep1.png" width="650"></div>
<p><em>The pcadapt Complete!</em></p>
</div>
</div>
<div id="outflank" class="section level2" number="6.2">
<h2>
<span class="header-section-number">6.2</span> OutFLANK<a class="anchor" aria-label="anchor" href="#outflank"><i class="fas fa-link"></i></a>
</h2>
<p>A Fst-based approach detects selection signals by comparing genetic
differentiation between defined group assignments <span class="citation">(<a href="sec-index.html#ref-whitlock2015">Whitlock and Lotterhos 2015</a>)</span>. For
more information, visit
<a href="https://rpubs.com/lotterhos/outflank" target="_blank">https://rpubs.com/lotterhos/outflank</a>.</p>
<div id="required-datasets-3" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Datasets:<a class="anchor" aria-label="anchor" href="#required-datasets-3"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<span style="color: #8b0000;"><strong><code>genlight</code></strong></span> with ‘Group Info.’,
downloadable from <u>Data Transform</u> page after you have
Group Info.</li>
<li>
<strong>Site Info.</strong> <strong>(RDS)</strong> of the current
<span style="color: #8b0000;"><strong><code>genlight</code></strong></span> downloadable from <u>Data
Input</u> or <u>Data QC</u> pages.</li>
</ul>
<hr>
</div>
<div id="steps-8" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-8"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Site Info.</strong>
(required).</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run
OutFLANK</strong></span> button
to perform genome scan for selection.</p></li>
</ol>
<hr>
</div>
<div id="outputs-17" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-17"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>OutFLANK</strong> <strong>p-value per site (RDS)</strong>: A dataset containing
p-values and adjusted p-values for each site.</p></li>
<li><p><strong>OutFLANK</strong> <strong>Manhattan Plot (PDF)</strong>: A Manhattan plot visualizing
the p-values per site across the genome. Significant SNPs are
highlighted in red.</p></li>
<li><p><strong>OutFLANK</strong> <strong>QQ Plot (PDF)</strong>: A QQ plot comparing the distribution
of observed p-values to the expected distribution under the null
hypothesis.</p></li>
<li><p><strong>OutFLANK</strong> <strong>Histogram of p-values (PDF)</strong>: A histogram showing
the distribution of p-values across all sites.</p></li>
<li><p><strong>OutFLANK</strong> <strong>Histogram of Fst (PDF)</strong>: A histogram showing the
distribution of Fst values across all sites.</p></li>
<li><p><strong>OutFLANK</strong> <strong>Significant SNPs (CSV)</strong>: A table listing SNPs
identified as significant by OutFLANK, including their site info.,
Fst values, and p-values.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/SelectionSweep2.png" width="650"></div>
<p><em>The OutFLANK</em> <em>Complete!</em></p>
<hr>
</div>
</div>
<div id="ibs-identity-by-state" class="section level2" number="6.3">
<h2>
<span class="header-section-number">6.3</span> IBS (Identity By State)<a class="anchor" aria-label="anchor" href="#ibs-identity-by-state"><i class="fas fa-link"></i></a>
</h2>
<p>An approach to detect differences in genomic regions between pairs of
individuals, useful for identifying pedigree relationships.</p>
<div id="required-datasets-4" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Datasets:<a class="anchor" aria-label="anchor" href="#required-datasets-4"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
<li>
<strong>Site Info.</strong> <strong>(RDS)</strong> of the current
<span style="color: #8b0000;"><strong><code>data.frame</code></strong></span>, downloadable from <u>Data
Input</u> or <u>Data QC</u> pages.</li>
<li>
<strong>Chromosome Info.</strong> <strong>(CSV)</strong>: Reference genome information of the
current study. For more details about this file, refer to
<strong>Section</strong> <a href="sec-data-qc.html#snp-density">2.3</a> <strong>(SNP Density)</strong>.</li>
</ul>
<hr>
</div>
<div id="steps-9" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-9"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Site Info.</strong>
(required).</p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Chromosome Info.</strong>
<strong>(CSV)</strong> (required).</p></li>
<li><p>Choose the reference and comparison samples.</p></li>
<li><p>Select window size (kb) and step size (kp).</p></li>
<li><p>To remove heterozygous SNPs from the reference sample, click the
<span style="color: #fff;background-color: #007ACC;">Remove heterozygous
SNPs</span> checkbox
(optional).</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run
IBS</strong></span> button to
perform IBS analysis.</p></li>
</ol>
<hr>
</div>
<div id="outputs-18" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-18"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>Chromosome Ideogram (PDF)</strong>: An ideogram visualizing the IBS
results, using a gradient palette to represent the differences
across chromosomes.</p></li>
<li><p><strong>Sliding Window Data (CSV)</strong>: A sliding window dataset with IBS
results, including SNP count, different SNPs, and the ratio of
different SNPs per window.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/SelectionSweep3.png" width="650"></div>
<p><em>The IBS Complete!</em></p>
<hr>
</div>
</div>
<div id="manhattan-plot-plus" class="section level2" number="6.4">
<h2>
<span class="header-section-number">6.4</span> Manhattan Plot <sup>Plus</sup><a class="anchor" aria-label="anchor" href="#manhattan-plot-plus"><i class="fas fa-link"></i></a>
</h2>
<p>Customize your phylogenetic tree plot based on the results from <u>Genetic
Diversity</u>/<u>Diversity Parameter</u>, <u>Selection
Sweep</u>/<u>pcadapt</u>, or <u>Selection
Sweep</u>/<u>OutFLANK</u>.</p>
<div id="required-files-2" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Files:<a class="anchor" aria-label="anchor" href="#required-files-2"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<strong>Genetic Diversity per Site</strong> (Genetic_Diversity_per_Site.rds),
<strong>pcadapt p-value per Site</strong> (pcadapt_p-value_per_site.rds), or
<strong>OutFLANK p-value per Site</strong> (OutFLANK_p-value_per_site.rds).</li>
<li>
<strong>Chromosome Info.</strong> <strong>(CSV)</strong>: Reference genome information of the
current study. For more details about this file, refer to
<strong>Section</strong> <a href="sec-data-qc.html#snp-density">2.3</a> <strong>(SNP Density)</strong>.</li>
</ul>
<hr>
</div>
<div id="steps-10" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-10"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span>
<strong>genetic_diversity/pcadapt_pvalue/OutFLANK_pvalue per site (RDS)</strong>.</p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Chromosome Info.
(CSV)</strong>.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Manhattan
Plot</strong></span> button to
generate the Manhattan plot.</p></li>
<li><p>Customize the Manhattan plot and click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Manhattan
Plot</strong></span> button
again.</p></li>
</ol>
<hr>
</div>
<div id="outputs-19" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-19"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<strong>Manhattan</strong> <strong>Plot (PDF)</strong>: A Manhattan plot with user-defined
layout style and attributes.</li>
</ul>
<blockquote>
<p><strong>Note</strong>: If generating a plot for p-values, make sure to use ‘-log10’
transformation for the Y axis.</p>
</blockquote>
<div class="inline-figure"><img src="screenshots/SelectionSweep4.png" width="650"></div>
<p><em>Manhattan</em> <em>Plot <sup>Plus</sup> Complete!</em></p>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></div>
<div class="next"><a href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-selection-sweep"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="nav-link" href="#pcadapt"><span class="header-section-number">6.1</span> pcadapt</a></li>
<li><a class="nav-link" href="#outflank"><span class="header-section-number">6.2</span> OutFLANK</a></li>
<li><a class="nav-link" href="#ibs-identity-by-state"><span class="header-section-number">6.3</span> IBS (Identity By State)</a></li>
<li><a class="nav-link" href="#manhattan-plot-plus"><span class="header-section-number">6.4</span> Manhattan Plot Plus</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/07-SelectionSweep.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/07-SelectionSweep.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on Feb 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
