<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>5 Genetic Diversity | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="➡️ This section contains four subpages: Diversity Parameter, Circos Plot, Genetic Distance, and AMOVA, allowing you to conduct various population diversity and differentiation analyses.   5.1...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="5 Genetic Diversity | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-genetic-diversity.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="➡️ This section contains four subpages: Diversity Parameter, Circos Plot, Genetic Distance, and AMOVA, allowing you to conduct various population diversity and differentiation analyses.   5.1...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="5 Genetic Diversity | User Guide for ShiNyP">
<meta name="twitter:description" content="➡️ This section contains four subpages: Diversity Parameter, Circos Plot, Genetic Distance, and AMOVA, allowing you to conduct various population diversity and differentiation analyses.   5.1...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Lato-0.4.9/font.css" rel="stylesheet">
<link href="libs/Roboto_Mono-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="" href="sec-quickstart.html">Quickstart</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="active" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="" href="sec-index.html"><span class="header-section-number">9</span> INDEX</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-genetic-diversity" class="section level1" number="5">
<h1>
<span class="header-section-number">5</span> Genetic Diversity<a class="anchor" aria-label="anchor" href="#sec-genetic-diversity"><i class="fas fa-link"></i></a>
</h1>
<div class="highlighted-text" style="background-color: #7c6f41; color: white; padding: 10px 5px; margin: 10px 0px; border-radius: 15px;">
<p>➡️ This section contains four subpages: <u><strong>Diversity
Parameter</strong></u>, <u><strong>Circos Plot</strong></u>, <u><strong>Genetic
Distance</strong></u>, and <u><strong>AMOVA</strong></u>, allowing you to
conduct various population diversity and differentiation analyses.</p>
</div>
<div class="inline-figure"><img src="images/Supp.%20Fig.%201-5_%E9%A0%81%E9%9D%A2_3.jpg" width="650"></div>
<div id="diversity-parameter" class="section level2" number="5.1">
<h2>
<span class="header-section-number">5.1</span> Diversity Parameter<a class="anchor" aria-label="anchor" href="#diversity-parameter"><i class="fas fa-link"></i></a>
</h2>
<p>Calculate key diversity parameters for each SNP site. This approach is
performed using the function from <em>snpReady</em> package <span class="citation">(<a href="sec-index.html#ref-granato2018">Granato et al. 2018</a>)</span>.</p>
<div id="required-datasets" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Datasets:<a class="anchor" aria-label="anchor" href="#required-datasets"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
<li>
<strong>Site Info.</strong> <strong>(RDS)</strong> of the current <strong><code>data.frame</code></strong>,
downloadable from <u>Data Input</u> or <u>Data QC</u>
pages.</li>
</ul>
<blockquote>
<p><strong>Note:</strong> If you upload Group Info., ensure that each group contains
at least 2 samples.</p>
</blockquote>
<hr>
</div>
<div id="steps-5" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-5"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Site Info.</strong>
(required).</p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Group Info.</strong> from
DAPC (optional). If uploaded, population-based parameters will be
calculated.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Diversity
Analysis</strong></span> button
to generate genetic diversity and the following downloadable files.</p></li>
</ol>
<hr>
</div>
<div id="outputs-12" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-12"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>Plot of Genetic Diversity Statistics per Site (PDF)</strong>: A
genome-wide scatter plot visualizing the user-selected parameter.</p></li>
<li><p><strong>Plot of Genetic Diversity Statistics by Group (PDF)</strong>: A lollipop
plot visualizing the user-selected parameter.</p></li>
<li><p><strong>Genetic Diversity per Site (RDS)</strong>: Contains site information and
diversity statistics, can be used as input data in the <u>Selection
Sweep</u>/<u>Manhattan Plot</u><sup>Plus</sup>.</p></li>
<li><p><strong>Genetic Diversity Object (RDS)</strong>: Contains all genetic diversity
results for future use and reproducibility.</p></li>
<li><p><strong>Genetic Diversity by Group</strong> <strong>(CSV)</strong>: A table showing genetic
diversity based on defined group assignments.</p></li>
<li><p><strong>Fst Matrix (CSV)</strong>: A table showing pairwise Fst based on defined
group assignments.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/GeneticDiversity1.png" width="650"></div>
<p><em>Diversity Analysis Complete!</em></p>
<hr>
</div>
</div>
<div id="circos-plot" class="section level2" number="5.2">
<h2>
<span class="header-section-number">5.2</span> Circos Plot<a class="anchor" aria-label="anchor" href="#circos-plot"><i class="fas fa-link"></i></a>
</h2>
<p>Genome-wide diversity is visualized using Circos plots generated with
the <em>circlize</em> package <span class="citation">(<a href="sec-index.html#ref-gu2014">Gu et al. 2014</a>)</span> based on results of diversity
parameters in a sliding window format.</p>
<div id="sec-required-dataset-chr" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#sec-required-dataset-chr"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p>Auto-import the results from the <u>Genetic
Diversity</u>/<u>Diversity Parameter</u> subpage.</p></li>
<li><p><strong>Chromosome Info.</strong> <strong>(CSV)</strong>: Reference genome information of the
current study. For more details about this file, refer to
<strong>Section</strong> <a href="sec-data-qc.html#snp-density">2.3</a> <strong>(SNP Density)</strong>.</p></li>
</ul>
<blockquote>
<p><strong>Note:</strong> Please ensure that each chromosome contains at least one SNP
marker.</p>
</blockquote>
<hr>
</div>
<div id="step-1-sliding-window" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Step 1: Sliding Window</strong><a class="anchor" aria-label="anchor" href="#step-1-sliding-window"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p>Select parameters to generate sliding window data.</p></li>
<li><p>Choose window size (kb) and step size (kp).</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Sliding
Window</strong></span> button to
generate sliding window data for circos plot.</p></li>
</ol>
<hr>
</div>
<div id="step-2-circos-plot" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Step 2: Circos Plot</strong><a class="anchor" aria-label="anchor" href="#step-2-circos-plot"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Chromosome Info.
(CSV)</strong>.</p></li>
<li><p>Select a parameter for each track, and add tracks if necessary (up
to a maximum of 6).</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Circos
Plot</strong></span> button to
generate the circos plot.</p></li>
</ol>
<hr>
</div>
<div id="outputs-13" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-13"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>Sliding Window Data (CSV)</strong>: A sliding window dataset based on
user-selected parameters.</p></li>
<li><p><strong>Circos Plot (PDF)</strong>: A circos plot visualizing the user-selected
parameters, with the top 1% of each parameter colored in red.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/GeneticDiversity2.png" width="650"></div>
<p><em>Circos Plot Complete!</em></p>
<hr>
</div>
</div>
<div id="genetic-distance" class="section level2" number="5.3">
<h2>
<span class="header-section-number">5.3</span> Genetic Distance<a class="anchor" aria-label="anchor" href="#genetic-distance"><i class="fas fa-link"></i></a>
</h2>
<p>Pairwise genetic distance between populations is computed using
<em>hierfstat</em> package. For more information, visit
<a href="https://rdrr.io/cran/hierfstat/man/genet.dist.html" target="_blank">https://rdrr.io/cran/hierfstat/man/genet.dist.html</a>.</p>
<div id="required-dataset-6" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#required-dataset-6"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<span style="color: #8b0000;"><strong><code>genind</code></strong></span> with ‘Group Info.’,
downloadable from <u>Data Conversion</u> page after you have
both the <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> and Group Info.</li>
</ul>
<hr>
</div>
<div id="steps-6" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Steps:</strong><a class="anchor" aria-label="anchor" href="#steps-6"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p>Select a method.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Genetic
Distance</strong></span> button
to generate the pairwise genetic distance.</p></li>
</ol>
<hr>
</div>
<div id="outputs-14" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-14"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>Genetic Distance Plot (PDF)</strong>: A plot of the pairwise genetic
distance matrix based on the user-selected method.</p></li>
<li><p><strong>Genetic Distance Matrix (CSV)</strong>: A pairwise genetic distance
matrix based on the user-selected method.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/GeneticDiversity3.png" width="650"></div>
<p><em>Genetic Distance Complete!</em></p>
<hr>
</div>
</div>
<div id="amova-analysis-of-molecular-variance" class="section level2" number="5.4">
<h2>
<span class="header-section-number">5.4</span> AMOVA (Analysis of MOlecular VAriance)<a class="anchor" aria-label="anchor" href="#amova-analysis-of-molecular-variance"><i class="fas fa-link"></i></a>
</h2>
<p>A method for assessing genetic variations and relationships within and
between populations <span class="citation">(<a href="sec-index.html#ref-excoffier1992">Excoffier, Smouse, and Quattro 1992</a>)</span>.This approach is performed using
the function from <em>hierfstat</em> and <em>poppr</em> packages <span class="citation">(<a href="sec-index.html#ref-kamvar2014">Kamvar, Tabima, and Grünwald 2014</a>; <a href="sec-index.html#ref-goudet2004">GOUDET 2004</a>)</span>.</p>
<div id="required-dataset-7" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required Dataset:<a class="anchor" aria-label="anchor" href="#required-dataset-7"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<span style="color: #8b0000;"><strong><code>genind</code></strong></span> with ‘Group Info.’,
downloadable from <u>Data Conversion</u> page after you have
both the <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> and Group Info.</li>
</ul>
<hr>
</div>
<div id="step-1-run-amova" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Step 1: Run AMOVA</strong><a class="anchor" aria-label="anchor" href="#step-1-run-amova"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run
AMOVA</strong></span> button to
partition genetic variation among and within populations.</li>
</ol>
<hr>
</div>
<div id="step-2-run-permutation-test" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>
<strong>Step 2: Run</strong> Permutation Test<a class="anchor" aria-label="anchor" href="#step-2-run-permutation-test"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li><p>Choose the number of randomizations for the permutation test to
detect the significance of three hierarchical levels. We recommend
using 9, 99 (default), 199, 499, 799, or 999 permutations for more
classical <em>p</em>-values.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Permutation
Test</strong></span> button to
perform the statistical test.</p></li>
</ol>
<hr>
</div>
<div id="outputs-15" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-15"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li><p><strong>AMOVA Variance Plot (PDF)</strong>: A pie chart showing the explained
genetic variance of population strata among defined groups.</p></li>
<li><p><strong>AMOVA Variance Test (PDF)</strong>: A plot showing the significance test
of population strata among defined groups. The histograms depict
randomized strata distributions, with the black line representing
genetic variance components.</p></li>
<li><p><strong>AMOVA Table (CSV)</strong>: A table with detailed AMOVA results.</p></li>
</ul>
<div class="inline-figure"><img src="screenshots/GeneticDiversity4.png" width="650"></div>
<p><em>AMOVA Complete!</em></p>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></div>
<div class="next"><a href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-genetic-diversity"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="nav-link" href="#diversity-parameter"><span class="header-section-number">5.1</span> Diversity Parameter</a></li>
<li><a class="nav-link" href="#circos-plot"><span class="header-section-number">5.2</span> Circos Plot</a></li>
<li><a class="nav-link" href="#genetic-distance"><span class="header-section-number">5.3</span> Genetic Distance</a></li>
<li><a class="nav-link" href="#amova-analysis-of-molecular-variance"><span class="header-section-number">5.4</span> AMOVA (Analysis of MOlecular VAriance)</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/06-GeneticDiversity.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/06-GeneticDiversity.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on Feb 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
