<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 Selection Sweep | ShiNyP: User Guide</title>
  <meta name="description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="8 Selection Sweep | ShiNyP: User Guide" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  <meta name="github-repo" content="TeddYenn/ShiNyP" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 Selection Sweep | ShiNyP: User Guide" />
  
  <meta name="twitter:description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sec-genetic-diversity.html"/>
<link rel="next" href="sec-core-collection.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ShiNyP: User Guide</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Quickstart</a></li>
<li class="chapter" data-level="2" data-path="sec-shinyp.html"><a href="sec-shinyp.html"><i class="fa fa-check"></i><b>2</b> <em>ShiNyP</em></a></li>
<li class="chapter" data-level="3" data-path="sec-data-input.html"><a href="sec-data-input.html"><i class="fa fa-check"></i><b>3</b> Data Input</a>
<ul>
<li class="chapter" data-level="3.1" data-path="sec-data-input.html"><a href="sec-data-input.html#vcf"><i class="fa fa-check"></i><b>3.1</b> VCF</a></li>
<li class="chapter" data-level="3.2" data-path="sec-data-input.html"><a href="sec-data-input.html#data.framegenindgenlight"><i class="fa fa-check"></i><b>3.2</b> data.frame/genind/genlight</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sec-data-qc.html"><a href="sec-data-qc.html"><i class="fa fa-check"></i><b>4</b> Data QC</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sec-data-qc.html"><a href="sec-data-qc.html#sample-qc"><i class="fa fa-check"></i><b>4.1</b> Sample QC</a></li>
<li class="chapter" data-level="4.2" data-path="sec-data-qc.html"><a href="sec-data-qc.html#snp-qc"><i class="fa fa-check"></i><b>4.2</b> SNP QC</a></li>
<li class="chapter" data-level="4.3" data-path="sec-data-qc.html"><a href="sec-data-qc.html#snp-density"><i class="fa fa-check"></i><b>4.3</b> SNP Density</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sec-data-conversion.html"><a href="sec-data-conversion.html"><i class="fa fa-check"></i><b>5</b> Data Transform</a></li>
<li class="chapter" data-level="6" data-path="sec-population-structure.html"><a href="sec-population-structure.html"><i class="fa fa-check"></i><b>6</b> Population Structure</a>
<ul>
<li class="chapter" data-level="6.1" data-path="sec-population-structure.html"><a href="sec-population-structure.html#pca-principal-component-analysis"><i class="fa fa-check"></i><b>6.1</b> PCA (Principal Component Analysis)</a></li>
<li class="chapter" data-level="6.2" data-path="sec-population-structure.html"><a href="sec-population-structure.html#dapc-discriminant-analysis-of-principal-components"><i class="fa fa-check"></i><b>6.2</b> DAPC (Discriminant Analysis of Principal Components)</a></li>
<li class="chapter" data-level="6.3" data-path="sec-population-structure.html"><a href="sec-population-structure.html#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree"><i class="fa fa-check"></i><b>6.3</b> UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree</a></li>
<li class="chapter" data-level="6.4" data-path="sec-population-structure.html"><a href="sec-population-structure.html#nj-neighbor-joining-tree"><i class="fa fa-check"></i><b>6.4</b> NJ (Neighbor-Joining) Tree</a></li>
<li class="chapter" data-level="6.5" data-path="sec-population-structure.html"><a href="sec-population-structure.html#kinship-analysis"><i class="fa fa-check"></i><b>6.5</b> Kinship Analysis</a></li>
<li class="chapter" data-level="6.6" data-path="sec-population-structure.html"><a href="sec-population-structure.html#scatter-plot-plus"><i class="fa fa-check"></i><b>6.6</b> Scatter Plot <sup>Plus</sup></a></li>
<li class="chapter" data-level="6.7" data-path="sec-population-structure.html"><a href="sec-population-structure.html#tree-plot-plus"><i class="fa fa-check"></i><b>6.7</b> Tree Plot <sup>Plus</sup></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html"><i class="fa fa-check"></i><b>7</b> Genetic Diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#diversity-parameter"><i class="fa fa-check"></i><b>7.1</b> Diversity Parameter</a></li>
<li class="chapter" data-level="7.2" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#circos-plot"><i class="fa fa-check"></i><b>7.2</b> Circos Plot</a></li>
<li class="chapter" data-level="7.3" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#genetic-distance"><i class="fa fa-check"></i><b>7.3</b> Genetic Distance</a></li>
<li class="chapter" data-level="7.4" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#amova-analysis-of-molecular-variance"><i class="fa fa-check"></i><b>7.4</b> AMOVA (Analysis of MOlecular VAriance)</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html"><i class="fa fa-check"></i><b>8</b> Selection Sweep</a>
<ul>
<li class="chapter" data-level="8.1" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#pcadapt"><i class="fa fa-check"></i><b>8.1</b> pcadapt</a></li>
<li class="chapter" data-level="8.2" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#outflank"><i class="fa fa-check"></i><b>8.2</b> OutFLANK</a></li>
<li class="chapter" data-level="8.3" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#ibs-identity-by-state"><i class="fa fa-check"></i><b>8.3</b> IBS (Identity By State)</a></li>
<li class="chapter" data-level="8.4" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#manhattan-plot-plus"><i class="fa fa-check"></i><b>8.4</b> Manhattan Plot <sup>Plus</sup></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="sec-core-collection.html"><a href="sec-core-collection.html"><i class="fa fa-check"></i><b>9</b> Core Collection</a>
<ul>
<li class="chapter" data-level="9.1" data-path="sec-core-collection.html"><a href="sec-core-collection.html#core-sample-set"><i class="fa fa-check"></i><b>9.1</b> Core Sample Set</a></li>
<li class="chapter" data-level="9.2" data-path="sec-core-collection.html"><a href="sec-core-collection.html#core-snp-set"><i class="fa fa-check"></i><b>9.2</b> Core SNP Set</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sec-ai-report.html"><a href="sec-ai-report.html"><i class="fa fa-check"></i><b>10</b> AI Report</a></li>
<li class="chapter" data-level="11" data-path="sec-index.html"><a href="sec-index.html"><i class="fa fa-check"></i><b>11</b> INDEX</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ShiNyP: User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sec-selection-sweep" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">8</span> Selection Sweep<a href="sec-selection-sweep.html#sec-selection-sweep" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="highlighted-text" style="padding-top: 15px; padding-bottom: 1px; padding-left: 15px">
<p>➡️ This section contains four subpages: <u><strong>pcadapt</strong></u>, <u><strong>OutFLANK</strong></u>, <u><strong>IBS</strong></u>, and <u><strong>Manhattan Plot</strong></u><sup><strong>Plus</strong></sup>, allowing you to detect selection signatures in different scenario and customize your plot.</p>
</div>
<p><img src="images/Supp.%20Fig.%201-5_%E9%A0%81%E9%9D%A2_4.jpg" /></p>
<div id="pcadapt" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> pcadapt<a href="sec-selection-sweep.html#pcadapt" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A PCA-based approach identifies selective outliers relative to population structure <span class="citation">(<a href="#ref-Luu2017">Luu, Bazin, and Blum 2016</a>)</span>.</p>
<div id="required-datasets-1" class="section level5 unnumbered hasAnchor">
<h5>Required Datasets:<a href="sec-selection-sweep.html#required-datasets-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
<li><strong>Site Info.</strong> <strong>(RDS)</strong> of the current <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span>, downloadable from <u>Data Input</u> or <u>Data QC</u> pages.</li>
</ul>
</div>
<div id="steps-7" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-selection-sweep.html#steps-7" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Site Info.</strong> (required).</p></li>
<li><p>Click <span style="color: #fff;background-color: #007ACC;">SNP Thinning</span> button (optional) and choose window size (number of SNPs) and r² threshold. For more information, visit <a href="https://bcm-uga.github.io/pcadapt/articles/pcadapt.html" target="_blank">https://bcm-uga.github.io/pcadapt/articles/pcadapt.html</a>.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run pcadapt</strong></span> button to perform genome scan for selection.</p></li>
</ol>
</div>
<div id="outputs-16" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-selection-sweep.html#outputs-16" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>pcadapt p-value per site (RDS)</strong>: A dataset containing p-values and adjusted p-values for each site.</p></li>
<li><p><strong>pcadapt Manhattan Plot (PDF)</strong>: A Manhattan plot visualizing the p-values per site across the genome. Significant SNPs are highlighted in red.</p></li>
<li><p><strong>pcadapt QQ Plot (PDF)</strong>: A QQ plot comparing the distribution of observed p-values to the expected distribution under the null hypothesis.</p></li>
<li><p><strong>pcadapt Histogram of p-values (PDF)</strong>: A histogram showing the distribution of p-values across all sites.</p></li>
<li><p><strong>pcadapt Histogram of Test Statistics (PDF)</strong>: A histogram showing the distribution of test statistics across all sites.</p></li>
<li><p><strong>pcadapt Significant SNPs (CSV)</strong>: A table listing SNPs identified as significant by pcadapt, including their site info., p-values, and adjusted p-values.</p></li>
</ul>
<p><img src="images/clipboard-1914990218.png" /></p>
<p><em>The pcadapt Complete!</em></p>
</div>
</div>
<div id="outflank" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> OutFLANK<a href="sec-selection-sweep.html#outflank" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A Fst-based approach detects selection signals by comparing genetic differentiation between defined group assignments <span class="citation">(<a href="#ref-whitlock2015">Whitlock and Lotterhos 2015</a>)</span>. For more information, visit <a href="https://rpubs.com/lotterhos/outflank" target="_blank">https://rpubs.com/lotterhos/outflank</a>.</p>
<div id="required-datasets-2" class="section level5 unnumbered hasAnchor">
<h5>Required Datasets:<a href="sec-selection-sweep.html#required-datasets-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>genind</code></strong></span> with ‘Group Info.’, downloadable from <u>Data Conversion</u> page after you have both the <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> and Group Info.</li>
<li><strong>Site Info.</strong> <strong>(RDS)</strong> of the current <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span>, downloadable from <u>Data Input</u> or <u>Data QC</u> pages.</li>
</ul>
</div>
<div id="steps-8" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-selection-sweep.html#steps-8" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Site Info.</strong> (required).</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run OutFLANK</strong></span> button to perform genome scan for selection.</p></li>
</ol>
</div>
<div id="outputs-17" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-selection-sweep.html#outputs-17" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>OutFLANK</strong> <strong>p-value per site (RDS)</strong>: A dataset containing p-values and adjusted p-values for each site.</p></li>
<li><p><strong>OutFLANK</strong> <strong>Manhattan Plot (PDF)</strong>: A Manhattan plot visualizing the p-values per site across the genome. Significant SNPs are highlighted in red.</p></li>
<li><p><strong>OutFLANK</strong> <strong>QQ Plot (PDF)</strong>: A QQ plot comparing the distribution of observed p-values to the expected distribution under the null hypothesis.</p></li>
<li><p><strong>OutFLANK</strong> <strong>Histogram of p-values (PDF)</strong>: A histogram showing the distribution of p-values across all sites.</p></li>
<li><p><strong>OutFLANK</strong> <strong>Histogram of Fst (PDF)</strong>: A histogram showing the distribution of Fst values across all sites.</p></li>
<li><p><strong>OutFLANK</strong> <strong>Significant SNPs (CSV)</strong>: A table listing SNPs identified as significant by OutFLANK, including their site info., Fst values, and p-values.</p></li>
</ul>
<p><img src="images/clipboard-1859743608.png" /></p>
<p><em>The OutFLANK</em> <em>Complete!</em></p>
</div>
</div>
<div id="ibs-identity-by-state" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> IBS (Identity By State)<a href="sec-selection-sweep.html#ibs-identity-by-state" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>An approach to detect differences in genomic regions between pairs of individuals, useful for identifying pedigree relationships.</p>
<div id="required-datasets-3" class="section level5 unnumbered hasAnchor">
<h5>Required Datasets:<a href="sec-selection-sweep.html#required-datasets-3" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
<li><strong>Site Info.</strong> <strong>(RDS)</strong> of the current <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span>, downloadable from <u>Data Input</u> or <u>Data QC</u> pages.</li>
<li><strong>Chromosome Info.</strong> <strong>(CSV)</strong>: Reference genome information of the current study. For more details about this file, refer to <strong>Section</strong> <a href="sec-data-qc.html#snp-density">4.3</a> <strong>(SNP Density)</strong>.</li>
</ul>
</div>
<div id="steps-9" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-selection-sweep.html#steps-9" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Site Info.</strong> (required).</p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Chromosome Info.</strong> <strong>(CSV)</strong> (required).</p></li>
<li><p>Choose the reference and comparison samples.</p></li>
<li><p>Select window size (kb) and step size (kp).</p></li>
<li><p>To remove heterozygous SNPs from the reference sample, click the <span style="color: #fff;background-color: #007ACC;">Remove heterozygous SNPs</span> checkbox (optional).</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run IBS</strong></span> button to perform IBS analysis.</p></li>
</ol>
</div>
<div id="outputs-18" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-selection-sweep.html#outputs-18" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>Chromosome Ideogram (PDF)</strong>: An ideogram visualizing the IBS results, using a gradient palette to represent the differences across chromosomes.</p></li>
<li><p><strong>Sliding Window Data (CSV)</strong>: A sliding window dataset with IBS results, including SNP count, different SNPs, and the ratio of different SNPs per window.</p></li>
</ul>
<p><img src="images/clipboard-4086295864.png" /></p>
<p><em>The IBS Complete!</em></p>
</div>
</div>
<div id="manhattan-plot-plus" class="section level2 hasAnchor" number="8.4">
<h2><span class="header-section-number">8.4</span> Manhattan Plot <sup>Plus</sup><a href="sec-selection-sweep.html#manhattan-plot-plus" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Customize your phylogenetic tree plot based on the results from <u>Genetic Diversity</u>/<u>Diversity Parameter</u>, <u>Selection Sweep</u>/<u>pcadapt</u>, or <u>Selection Sweep</u>/<u>OutFLANK</u>.</p>
<div id="required-files-2" class="section level5 unnumbered hasAnchor">
<h5>Required Files:<a href="sec-selection-sweep.html#required-files-2" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><strong>Genetic Diversity per Site</strong> (Genetic_Diversity_per_Site.rds), <strong>pcadapt p-value per Site</strong> (pcadapt_p-value_per_site.rds), or <strong>OutFLANK p-value per Site</strong> (OutFLANK_p-value_per_site.rds).</li>
<li><strong>Chromosome Info.</strong> <strong>(CSV)</strong>: Reference genome information of the current study. For more details about this file, refer to <strong>Section</strong> <a href="sec-data-qc.html#snp-density">4.3</a> <strong>(SNP Density)</strong>.</li>
</ul>
</div>
<div id="steps-10" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-selection-sweep.html#steps-10" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>genetic_diversity/pcadapt_pvalue/OutFLANK_pvalue per site (RDS)</strong>.</p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Chromosome Info. (CSV)</strong>.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Manhattan Plot</strong></span> button to generate the Manhattan plot.</p></li>
<li><p>Customize the Manhattan plot and click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Manhattan Plot</strong></span> button again.</p></li>
</ol>
</div>
<div id="outputs-19" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-selection-sweep.html#outputs-19" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><strong>Manhattan</strong> <strong>Plot (PDF)</strong>: A Manhattan plot with user-defined layout style and attributes.</li>
</ul>
<blockquote>
<p><strong>Note</strong>: If generating a plot for p-values, make sure to use ‘-log10’ transformation for the Y axis.</p>
</blockquote>
<p><img src="images/clipboard-2598995324.png" /></p>
<p><em>Manhattan</em> <em>Plot <sup>Plus</sup> Complete!</em></p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Luu2017" class="csl-entry">
Luu, Keurcien, Eric Bazin, and Michael G. B. Blum. 2016. <span>“<span><em>Pcadapt</em></span>: An R Package to Perform Genome Scans for Selection Based on Principal Component Analysis.”</span> <em>Molecular Ecology Resources</em> 17 (1): 67–77. <a href="https://doi.org/10.1111/1755-0998.12592">https://doi.org/10.1111/1755-0998.12592</a>.
</div>
<div id="ref-whitlock2015" class="csl-entry">
Whitlock, Michael C., and Katie E. Lotterhos. 2015. <span>“Reliable Detection of Loci Responsible for Local Adaptation: Inference of a Null Model Through Trimming the Distribution of<span><em>F</em></span><span><sub>ST</sub></span>.”</span> <em>The American Naturalist</em> 186 (S1): S24–36. <a href="https://doi.org/10.1086/682949">https://doi.org/10.1086/682949</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sec-genetic-diversity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sec-core-collection.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/07-SelectionSweep.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["ShiNyP-guide.pdf", "ShiNyP-guide.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
