<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Genetic Diversity | ShiNyP: User Guide</title>
  <meta name="description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Genetic Diversity | ShiNyP: User Guide" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  <meta name="github-repo" content="TeddYenn/ShiNyP" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Genetic Diversity | ShiNyP: User Guide" />
  
  <meta name="twitter:description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sec-population-structure.html"/>
<link rel="next" href="sec-selection-sweep.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ShiNyP: User Guide</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Quickstart</a></li>
<li class="chapter" data-level="2" data-path="sec-shinyp.html"><a href="sec-shinyp.html"><i class="fa fa-check"></i><b>2</b> <em>ShiNyP</em></a></li>
<li class="chapter" data-level="3" data-path="sec-data-input.html"><a href="sec-data-input.html"><i class="fa fa-check"></i><b>3</b> Data Input</a>
<ul>
<li class="chapter" data-level="3.1" data-path="sec-data-input.html"><a href="sec-data-input.html#vcf"><i class="fa fa-check"></i><b>3.1</b> VCF</a></li>
<li class="chapter" data-level="3.2" data-path="sec-data-input.html"><a href="sec-data-input.html#data.framegenindgenlight"><i class="fa fa-check"></i><b>3.2</b> data.frame/genind/genlight</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sec-data-qc.html"><a href="sec-data-qc.html"><i class="fa fa-check"></i><b>4</b> Data QC</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sec-data-qc.html"><a href="sec-data-qc.html#sample-qc"><i class="fa fa-check"></i><b>4.1</b> Sample QC</a></li>
<li class="chapter" data-level="4.2" data-path="sec-data-qc.html"><a href="sec-data-qc.html#snp-qc"><i class="fa fa-check"></i><b>4.2</b> SNP QC</a></li>
<li class="chapter" data-level="4.3" data-path="sec-data-qc.html"><a href="sec-data-qc.html#snp-density"><i class="fa fa-check"></i><b>4.3</b> SNP Density</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sec-data-conversion.html"><a href="sec-data-conversion.html"><i class="fa fa-check"></i><b>5</b> Data Transform</a></li>
<li class="chapter" data-level="6" data-path="sec-population-structure.html"><a href="sec-population-structure.html"><i class="fa fa-check"></i><b>6</b> Population Structure</a>
<ul>
<li class="chapter" data-level="6.1" data-path="sec-population-structure.html"><a href="sec-population-structure.html#pca-principal-component-analysis"><i class="fa fa-check"></i><b>6.1</b> PCA (Principal Component Analysis)</a></li>
<li class="chapter" data-level="6.2" data-path="sec-population-structure.html"><a href="sec-population-structure.html#dapc-discriminant-analysis-of-principal-components"><i class="fa fa-check"></i><b>6.2</b> DAPC (Discriminant Analysis of Principal Components)</a></li>
<li class="chapter" data-level="6.3" data-path="sec-population-structure.html"><a href="sec-population-structure.html#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree"><i class="fa fa-check"></i><b>6.3</b> UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree</a></li>
<li class="chapter" data-level="6.4" data-path="sec-population-structure.html"><a href="sec-population-structure.html#nj-neighbor-joining-tree"><i class="fa fa-check"></i><b>6.4</b> NJ (Neighbor-Joining) Tree</a></li>
<li class="chapter" data-level="6.5" data-path="sec-population-structure.html"><a href="sec-population-structure.html#kinship-analysis"><i class="fa fa-check"></i><b>6.5</b> Kinship Analysis</a></li>
<li class="chapter" data-level="6.6" data-path="sec-population-structure.html"><a href="sec-population-structure.html#scatter-plot-plus"><i class="fa fa-check"></i><b>6.6</b> Scatter Plot <sup>Plus</sup></a></li>
<li class="chapter" data-level="6.7" data-path="sec-population-structure.html"><a href="sec-population-structure.html#tree-plot-plus"><i class="fa fa-check"></i><b>6.7</b> Tree Plot <sup>Plus</sup></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html"><i class="fa fa-check"></i><b>7</b> Genetic Diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#diversity-parameter"><i class="fa fa-check"></i><b>7.1</b> Diversity Parameter</a></li>
<li class="chapter" data-level="7.2" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#circos-plot"><i class="fa fa-check"></i><b>7.2</b> Circos Plot</a></li>
<li class="chapter" data-level="7.3" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#genetic-distance"><i class="fa fa-check"></i><b>7.3</b> Genetic Distance</a></li>
<li class="chapter" data-level="7.4" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#amova-analysis-of-molecular-variance"><i class="fa fa-check"></i><b>7.4</b> AMOVA (Analysis of MOlecular VAriance)</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html"><i class="fa fa-check"></i><b>8</b> Selection Sweep</a>
<ul>
<li class="chapter" data-level="8.1" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#pcadapt"><i class="fa fa-check"></i><b>8.1</b> pcadapt</a></li>
<li class="chapter" data-level="8.2" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#outflank"><i class="fa fa-check"></i><b>8.2</b> OutFLANK</a></li>
<li class="chapter" data-level="8.3" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#ibs-identity-by-state"><i class="fa fa-check"></i><b>8.3</b> IBS (Identity By State)</a></li>
<li class="chapter" data-level="8.4" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#manhattan-plot-plus"><i class="fa fa-check"></i><b>8.4</b> Manhattan Plot <sup>Plus</sup></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="sec-core-collection.html"><a href="sec-core-collection.html"><i class="fa fa-check"></i><b>9</b> Core Collection</a>
<ul>
<li class="chapter" data-level="9.1" data-path="sec-core-collection.html"><a href="sec-core-collection.html#core-sample-set"><i class="fa fa-check"></i><b>9.1</b> Core Sample Set</a></li>
<li class="chapter" data-level="9.2" data-path="sec-core-collection.html"><a href="sec-core-collection.html#core-snp-set"><i class="fa fa-check"></i><b>9.2</b> Core SNP Set</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sec-ai-report.html"><a href="sec-ai-report.html"><i class="fa fa-check"></i><b>10</b> AI Report</a></li>
<li class="chapter" data-level="11" data-path="sec-index.html"><a href="sec-index.html"><i class="fa fa-check"></i><b>11</b> INDEX</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ShiNyP: User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sec-genetic-diversity" class="section level1 hasAnchor" number="7">
<h1><span class="header-section-number">7</span> Genetic Diversity<a href="sec-genetic-diversity.html#sec-genetic-diversity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div class="highlighted-text" style="padding-top: 15px; padding-bottom: 1px; padding-left: 15px">
<p>➡️ This section contains four subpages: <u><strong>Diversity Parameter</strong></u>, <u><strong>Circos Plot</strong></u>, <u><strong>Genetic Distance</strong></u>, and <u><strong>AMOVA</strong></u>, allowing you to conduct various population diversity and differentiation analyses.</p>
</div>
<p><img src="images/Supp.%20Fig.%201-5_%E9%A0%81%E9%9D%A2_3.jpg" /></p>
<div id="diversity-parameter" class="section level2 hasAnchor" number="7.1">
<h2><span class="header-section-number">7.1</span> Diversity Parameter<a href="sec-genetic-diversity.html#diversity-parameter" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Calculate key diversity parameters for each SNP site. This approach is performed using the function from <em>snpReady</em> package <span class="citation">(<a href="#ref-granato2018">Granato et al. 2018</a>)</span>.</p>
<div id="required-datasets" class="section level5 unnumbered hasAnchor">
<h5>Required Datasets:<a href="sec-genetic-diversity.html#required-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>data.frame</code></strong></span></li>
<li><strong>Site Info.</strong> <strong>(RDS)</strong> of the current <strong><code>data.frame</code></strong>, downloadable from <u>Data Input</u> or <u>Data QC</u> pages.</li>
</ul>
</div>
<div id="steps-5" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-genetic-diversity.html#steps-5" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Site Info.</strong> (required).</p></li>
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Group Info.</strong> from DAPC (optional). If uploaded, population-based parameters will be calculated.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Diversity Analysis</strong></span> button to generate genetic diversity and the following downloadable files.</p></li>
</ol>
</div>
<div id="outputs-12" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-genetic-diversity.html#outputs-12" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>Plot of Genetic Diversity Statistics per Site (PDF)</strong>: A genome-wide scatter plot visualizing the user-selected parameter.</p></li>
<li><p><strong>Plot of Genetic Diversity Statistics by Group (PDF)</strong>: A lollipop plot visualizing the user-selected parameter.</p></li>
<li><p><strong>Genetic Diversity per Site (RDS)</strong>: Contains site information and diversity statistics, can be used as input data in the <u>Selection Sweep</u>/<u>Manhattan Plot</u><sup>Plus</sup>.</p></li>
<li><p><strong>Genetic Diversity Object (RDS)</strong>: Contains all genetic diversity results for future use and reproducibility.</p></li>
<li><p><strong>Genetic Diversity by Group</strong> <strong>(CSV)</strong>: A table showing genetic diversity based on defined group assignments.</p></li>
<li><p><strong>Fst Matrix (CSV)</strong>: A table showing pairwise Fst based on defined group assignments.</p></li>
</ul>
<p><img src="images/clipboard-4215023438.png" /></p>
<p><em>Diversity Analysis Complete!</em></p>
</div>
</div>
<div id="circos-plot" class="section level2 hasAnchor" number="7.2">
<h2><span class="header-section-number">7.2</span> Circos Plot<a href="sec-genetic-diversity.html#circos-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Genome-wide diversity is visualized using Circos plots generated with the <em>circlize</em> package <span class="citation">(<a href="#ref-gu2014">Gu et al. 2014</a>)</span> based on results of diversity parameters in a sliding window format.</p>
<div id="sec-required-dataset-chr" class="section level5 unnumbered hasAnchor">
<h5>Required Dataset:<a href="sec-genetic-diversity.html#sec-required-dataset-chr" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p>Auto-import the results from the <u>Genetic Diversity</u>/<u>Diversity Parameter</u> subpage.</p></li>
<li><p><strong>Chromosome Info.</strong> <strong>(CSV)</strong>: Reference genome information of the current study. For more details about this file, refer to <strong>Section</strong> <a href="sec-data-qc.html#snp-density">4.3</a> <strong>(SNP Density)</strong>.</p></li>
</ul>
</div>
<div id="step-1-sliding-window" class="section level5 unnumbered hasAnchor">
<h5><strong>Step 1: Sliding Window</strong><a href="sec-genetic-diversity.html#step-1-sliding-window" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p>Select parameters to generate sliding window data.</p></li>
<li><p>Choose window size (kb) and step size (kp).</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Sliding Window</strong></span> button to generate sliding window data for circos plot.</p></li>
</ol>
</div>
<div id="step-2-circos-plot" class="section level5 unnumbered hasAnchor">
<h5><strong>Step 2: Circos Plot</strong><a href="sec-genetic-diversity.html#step-2-circos-plot" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p><span style="background-color: #fceed2;">Upload</span> <strong>Chromosome Info. (CSV)</strong>.</p></li>
<li><p>Select a parameter for each track, and add tracks if necessary (up to a maximum of 6).</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Circos Plot</strong></span> button to generate the circos plot.</p></li>
</ol>
</div>
<div id="outputs-13" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-genetic-diversity.html#outputs-13" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>Sliding Window Data (CSV)</strong>: A sliding window dataset based on user-selected parameters.</p></li>
<li><p><strong>Circos Plot (PDF)</strong>: A circos plot visualizing the user-selected parameters, with the top 1% of each parameter colored in red.</p></li>
</ul>
<p><img src="images/clipboard-3157688358.png" /></p>
<p><em>Circos Plot Complete!</em></p>
</div>
</div>
<div id="genetic-distance" class="section level2 hasAnchor" number="7.3">
<h2><span class="header-section-number">7.3</span> Genetic Distance<a href="sec-genetic-diversity.html#genetic-distance" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Pairwise genetic distance between populations is computed using <em>hierfstat</em> package. For more information, visit <a href="#id_0">https://rdrr.io/cran/hierfstat/man/genet.dist.html</a>.</p>
<div id="required-dataset-6" class="section level5 unnumbered hasAnchor">
<h5>Required Dataset:<a href="sec-genetic-diversity.html#required-dataset-6" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>genind</code></strong></span> with ‘Group Info.’, downloadable from <u>Data Conversion</u> page after you have both the <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> and Group Info.</li>
</ul>
</div>
<div id="steps-6" class="section level5 unnumbered hasAnchor">
<h5><strong>Steps:</strong><a href="sec-genetic-diversity.html#steps-6" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p>Select a method.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Genetic Distance</strong></span> button to generate the pairwise genetic distance.</p></li>
</ol>
</div>
<div id="outputs-14" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-genetic-diversity.html#outputs-14" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>Genetic Distance Plot (PDF)</strong>: A plot of the pairwise genetic distance matrix based on the user-selected method.</p></li>
<li><p><strong>Genetic Distance Matrix (CSV)</strong>: A pairwise genetic distance matrix based on the user-selected method.</p></li>
</ul>
<p><img src="images/clipboard-124406020.png" /></p>
<p><em>Genetic Distance Complete!</em></p>
</div>
</div>
<div id="amova-analysis-of-molecular-variance" class="section level2 hasAnchor" number="7.4">
<h2><span class="header-section-number">7.4</span> AMOVA (Analysis of MOlecular VAriance)<a href="sec-genetic-diversity.html#amova-analysis-of-molecular-variance" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A method for assessing genetic variations and relationships within and between populations <span class="citation">(<a href="#ref-excoffier1992">Excoffier, Smouse, and Quattro 1992</a>)</span>.This approach is performed using the function from <em>hierfstat</em> and <em>poppr</em> packages <span class="citation">(<a href="#ref-kamvar2014">Kamvar, Tabima, and Grünwald 2014</a>; <a href="#ref-goudet2004">GOUDET 2004</a>)</span>.</p>
<div id="required-dataset-7" class="section level5 unnumbered hasAnchor">
<h5>Required Dataset:<a href="sec-genetic-diversity.html#required-dataset-7" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><span style="color: #8b0000;"><strong><code>genind</code></strong></span> with ‘Group Info.’, downloadable from <u>Data Conversion</u> page after you have both the <span style="color: #8b0000;"><strong><code>data.frame</code></strong></span> and Group Info.</li>
</ul>
</div>
<div id="step-1-run-amova" class="section level5 unnumbered hasAnchor">
<h5><strong>Step 1: Run AMOVA</strong><a href="sec-genetic-diversity.html#step-1-run-amova" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run AMOVA</strong></span> button to partition genetic variation among and within populations.</li>
</ol>
</div>
<div id="step-2-run-permutation-test" class="section level5 unnumbered hasAnchor">
<h5><strong>Step 2: Run</strong> Permutation Test<a href="sec-genetic-diversity.html#step-2-run-permutation-test" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ol style="list-style-type: decimal">
<li><p>Choose the number of randomizations for the permutation test to detect the significance of three hierarchical levels. We recommend using 9, 99 (default), 199, 499, 799, or 999 permutations for more classical <em>p</em>-values.</p></li>
<li><p>Click the <span style="color: #fff;background-color: #007ACC;"><strong>Run Permutation Test</strong></span> button to perform the statistical test.</p></li>
</ol>
</div>
<div id="outputs-15" class="section level5 unnumbered hasAnchor">
<h5>Outputs:<a href="sec-genetic-diversity.html#outputs-15" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><p><strong>AMOVA Variance Plot (PDF)</strong>: A pie chart showing the explained genetic variance of population strata among defined groups.</p></li>
<li><p><strong>AMOVA Variance Test (PDF)</strong>: A plot showing the significance test of population strata among defined groups. The histograms depict randomized strata distributions, with the black line representing genetic variance components.</p></li>
<li><p><strong>AMOVA Table (CSV)</strong>: A table with detailed AMOVA results.</p></li>
</ul>
<p><img src="images/clipboard-1689228022.png" /></p>
<p><em>AMOVA Complete!</em></p>

</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-excoffier1992" class="csl-entry">
Excoffier, L, P E Smouse, and J M Quattro. 1992. <span>“Analysis of Molecular Variance Inferred from Metric Distances Among DNA Haplotypes: Application to Human Mitochondrial DNA Restriction Data.”</span> <em>Genetics</em> 131 (2): 479–91. <a href="https://doi.org/10.1093/genetics/131.2.479">https://doi.org/10.1093/genetics/131.2.479</a>.
</div>
<div id="ref-goudet2004" class="csl-entry">
GOUDET, JÉRÔME. 2004. <span>“Hierfstat, a Package for r to Compute and Test Hierarchical <span> <em>F</em></span><span>-</span>Statistics.”</span> <em>Molecular Ecology Notes</em> 5 (1): 184–86. <a href="https://doi.org/10.1111/j.1471-8286.2004.00828.x">https://doi.org/10.1111/j.1471-8286.2004.00828.x</a>.
</div>
<div id="ref-granato2018" class="csl-entry">
Granato, Italo S. C., Giovanni Galli, Evellyn Giselly de Oliveira Couto, Massaine Bandeira e Souza, Leandro Freitas Mendonça, and Roberto Fritsche-Neto. 2018. <span>“snpReady: A Tool to Assist Breeders in Genomic Analysis.”</span> <em>Molecular Breeding</em> 38 (8). <a href="https://doi.org/10.1007/s11032-018-0844-8">https://doi.org/10.1007/s11032-018-0844-8</a>.
</div>
<div id="ref-gu2014" class="csl-entry">
Gu, Zuguang, Lei Gu, Roland Eils, Matthias Schlesner, and Benedikt Brors. 2014. <span>“<span><em>Circlize</em></span>implements and Enhances Circular Visualization in R.”</span> <em>Bioinformatics</em> 30 (19): 2811–12. <a href="https://doi.org/10.1093/bioinformatics/btu393">https://doi.org/10.1093/bioinformatics/btu393</a>.
</div>
<div id="ref-kamvar2014" class="csl-entry">
Kamvar, Zhian N., Javier F. Tabima, and Niklaus J. Grünwald. 2014. <span>“<span><em>Poppr</em></span>: An R Package for Genetic Analysis of Populations with Clonal, Partially Clonal, and/or Sexual Reproduction.”</span> <em>PeerJ</em> 2 (March): e281. <a href="https://doi.org/10.7717/peerj.281">https://doi.org/10.7717/peerj.281</a>.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sec-population-structure.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sec-selection-sweep.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/06-GeneticDiversity.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["ShiNyP-guide.pdf", "ShiNyP-guide.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
