<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>11 INDEX | ShiNyP: User Guide</title>
  <meta name="description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="11 INDEX | ShiNyP: User Guide" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  <meta name="github-repo" content="TeddYenn/ShiNyP" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="11 INDEX | ShiNyP: User Guide" />
  
  <meta name="twitter:description" content="A Shiny-Based Interactive Platform for Genome-Wide SNP Analysis and Visualization" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sec-ai-report.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>
<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ShiNyP: User Guide</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Quickstart</a></li>
<li class="chapter" data-level="2" data-path="sec-shinyp.html"><a href="sec-shinyp.html"><i class="fa fa-check"></i><b>2</b> <em>ShiNyP</em></a></li>
<li class="chapter" data-level="3" data-path="sec-data-input.html"><a href="sec-data-input.html"><i class="fa fa-check"></i><b>3</b> Data Input</a>
<ul>
<li class="chapter" data-level="3.1" data-path="sec-data-input.html"><a href="sec-data-input.html#vcf"><i class="fa fa-check"></i><b>3.1</b> VCF</a></li>
<li class="chapter" data-level="3.2" data-path="sec-data-input.html"><a href="sec-data-input.html#data.framegenindgenlight"><i class="fa fa-check"></i><b>3.2</b> data.frame/genind/genlight</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sec-data-qc.html"><a href="sec-data-qc.html"><i class="fa fa-check"></i><b>4</b> Data QC</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sec-data-qc.html"><a href="sec-data-qc.html#sample-qc"><i class="fa fa-check"></i><b>4.1</b> Sample QC</a></li>
<li class="chapter" data-level="4.2" data-path="sec-data-qc.html"><a href="sec-data-qc.html#snp-qc"><i class="fa fa-check"></i><b>4.2</b> SNP QC</a></li>
<li class="chapter" data-level="4.3" data-path="sec-data-qc.html"><a href="sec-data-qc.html#snp-density"><i class="fa fa-check"></i><b>4.3</b> SNP Density</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="sec-data-conversion.html"><a href="sec-data-conversion.html"><i class="fa fa-check"></i><b>5</b> Data Transform</a></li>
<li class="chapter" data-level="6" data-path="sec-population-structure.html"><a href="sec-population-structure.html"><i class="fa fa-check"></i><b>6</b> Population Structure</a>
<ul>
<li class="chapter" data-level="6.1" data-path="sec-population-structure.html"><a href="sec-population-structure.html#pca-principal-component-analysis"><i class="fa fa-check"></i><b>6.1</b> PCA (Principal Component Analysis)</a></li>
<li class="chapter" data-level="6.2" data-path="sec-population-structure.html"><a href="sec-population-structure.html#dapc-discriminant-analysis-of-principal-components"><i class="fa fa-check"></i><b>6.2</b> DAPC (Discriminant Analysis of Principal Components)</a></li>
<li class="chapter" data-level="6.3" data-path="sec-population-structure.html"><a href="sec-population-structure.html#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree"><i class="fa fa-check"></i><b>6.3</b> UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree</a></li>
<li class="chapter" data-level="6.4" data-path="sec-population-structure.html"><a href="sec-population-structure.html#nj-neighbor-joining-tree"><i class="fa fa-check"></i><b>6.4</b> NJ (Neighbor-Joining) Tree</a></li>
<li class="chapter" data-level="6.5" data-path="sec-population-structure.html"><a href="sec-population-structure.html#kinship-analysis"><i class="fa fa-check"></i><b>6.5</b> Kinship Analysis</a></li>
<li class="chapter" data-level="6.6" data-path="sec-population-structure.html"><a href="sec-population-structure.html#scatter-plot-plus"><i class="fa fa-check"></i><b>6.6</b> Scatter Plot <sup>Plus</sup></a></li>
<li class="chapter" data-level="6.7" data-path="sec-population-structure.html"><a href="sec-population-structure.html#tree-plot-plus"><i class="fa fa-check"></i><b>6.7</b> Tree Plot <sup>Plus</sup></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html"><i class="fa fa-check"></i><b>7</b> Genetic Diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#diversity-parameter"><i class="fa fa-check"></i><b>7.1</b> Diversity Parameter</a></li>
<li class="chapter" data-level="7.2" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#circos-plot"><i class="fa fa-check"></i><b>7.2</b> Circos Plot</a></li>
<li class="chapter" data-level="7.3" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#genetic-distance"><i class="fa fa-check"></i><b>7.3</b> Genetic Distance</a></li>
<li class="chapter" data-level="7.4" data-path="sec-genetic-diversity.html"><a href="sec-genetic-diversity.html#amova-analysis-of-molecular-variance"><i class="fa fa-check"></i><b>7.4</b> AMOVA (Analysis of MOlecular VAriance)</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html"><i class="fa fa-check"></i><b>8</b> Selection Sweep</a>
<ul>
<li class="chapter" data-level="8.1" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#pcadapt"><i class="fa fa-check"></i><b>8.1</b> pcadapt</a></li>
<li class="chapter" data-level="8.2" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#outflank"><i class="fa fa-check"></i><b>8.2</b> OutFLANK</a></li>
<li class="chapter" data-level="8.3" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#ibs-identity-by-state"><i class="fa fa-check"></i><b>8.3</b> IBS (Identity By State)</a></li>
<li class="chapter" data-level="8.4" data-path="sec-selection-sweep.html"><a href="sec-selection-sweep.html#manhattan-plot-plus"><i class="fa fa-check"></i><b>8.4</b> Manhattan Plot <sup>Plus</sup></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="sec-core-collection.html"><a href="sec-core-collection.html"><i class="fa fa-check"></i><b>9</b> Core Collection</a>
<ul>
<li class="chapter" data-level="9.1" data-path="sec-core-collection.html"><a href="sec-core-collection.html#core-sample-set"><i class="fa fa-check"></i><b>9.1</b> Core Sample Set</a></li>
<li class="chapter" data-level="9.2" data-path="sec-core-collection.html"><a href="sec-core-collection.html#core-snp-set"><i class="fa fa-check"></i><b>9.2</b> Core SNP Set</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sec-ai-report.html"><a href="sec-ai-report.html"><i class="fa fa-check"></i><b>10</b> AI Report</a></li>
<li class="chapter" data-level="11" data-path="sec-index.html"><a href="sec-index.html"><i class="fa fa-check"></i><b>11</b> INDEX</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ShiNyP: User Guide</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sec-index" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">11</span> INDEX<a href="sec-index.html#sec-index" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ul>
<li><p>AI Report <a href="https://teddyenn.github.io/ShiNyP-guide/sec-ai-report.html#sec-ai-report">10</a></p></li>
<li><p>AMOVA (Analysis of MOlecular VAriance) <a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#amova-analysis-of-molecular-variance">7.4</a></p></li>
<li><p>API key <a href="https://teddyenn.github.io/ShiNyP-guide/sec-ai-report.html#step-2-ai-driven-report">10</a></p></li>
<li><p>Bayesian Information Criterion (BIC) <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#step-2-dapc-analysis">6.2</a></p></li>
<li><p>Chromosome Info. <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-density">4.3</a></p></li>
<li><p>Circos Plot <a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#circos-plot">7.2</a></p></li>
<li><p>Core Sample Set <a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-sample-set">9.1</a></p></li>
<li><p>Core SNP Set <a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-snp-set">9.2</a></p></li>
<li><p>DAPC (Discriminant Analysis of Principal Components) <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#dapc-discriminant-analysis-of-principal-components">6.2</a></p></li>
<li><p>data.frame <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-2-transform-to-data.frame">3.1</a></p></li>
<li><p>Demo Data <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-1-input-your-vcf-file">3.1</a></p></li>
<li><p>Diversity Parameter <a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#diversity-parameter">7.1</a></p></li>
<li><p>Genetic Distance <a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#genetic-distance">7.3</a></p></li>
<li><p>genind <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html#step-1-transform-data.frame-to-genind">5</a></p></li>
<li><p>genlight <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html#step-2-transform-genind-to-genlight">5</a></p></li>
<li><p>Group Info. <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#scatter-plot-plus">6.6</a></p></li>
<li><p>Hardy-Weinberg equilibrium (HWE) <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">4.2</a></p></li>
<li><p>Heterozygosity rate <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">4.1</a></p></li>
<li><p>IBS (Identity By State) <a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#ibs-identity-by-state">8.3</a></p></li>
<li><p>Kinship Analysis <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#kinship-analysis">6.5</a></p></li>
<li><p>Manhattan Plot <a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#manhattan-plot-plus">8.4</a></p></li>
<li><p>Minor allele frequency (MAF) <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">4.2</a></p></li>
<li><p>Missing rate <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">4.1</a></p></li>
<li><p>NJ (Neighbor-Joining) Tree <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#nj-neighbor-joining-tree">6.4</a></p></li>
<li><p>OutFLANK <a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#outflank">8.2</a></p></li>
<li><p>PCA (Principal Component Analysis) <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#pca-principal-component-analysis">6.1</a></p></li>
<li><p>pcadapt <a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#pcadapt">8.1</a></p></li>
<li><p>Permutation Test <a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#step-2-run-permutation-test">7.4</a></p></li>
<li><p>Sample QC <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">4.1</a></p></li>
<li><p>Scatter Plot <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">6.6</a></p></li>
<li><p><em>ShiNyP</em> <a href="https://teddyenn.github.io/ShiNyP-guide/sec-shinyp.html#sec-shinyp">2</a></p></li>
<li><p>Site Info. <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-2-transform-to-data.frame">3.1</a></p></li>
<li><p>SNP Density <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-density">4.3</a></p></li>
<li><p>SNP QC <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">4.2</a></p></li>
<li><p>Tree Plot <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#tree-plot-plus">6.7</a></p></li>
<li><p>UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree">6.3</a></p></li>
<li><p>VCF <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#vcf">3.1</a></p></li>
</ul>
<hr />
<div id="bibliography" class="section level4 unnumbered hasAnchor">
<h4>Bibliography<a href="sec-index.html#bibliography" class="anchor-section" aria-label="Anchor link to header"></a></h4>

<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div class="csl-entry">
Excoffier, L, P E Smouse, and J M Quattro. 1992. <span>“Analysis of Molecular Variance Inferred from Metric Distances Among DNA Haplotypes: Application to Human Mitochondrial DNA Restriction Data.”</span> <em>Genetics</em> 131 (2): 479–91. <a href="https://doi.org/10.1093/genetics/131.2.479">https://doi.org/10.1093/genetics/131.2.479</a>.
</div>
<div class="csl-entry">
GOUDET, JÉRÔME. 2004. <span>“Hierfstat, a Package for r to Compute and Test Hierarchical <span> <em>F</em></span><span>-</span>Statistics.”</span> <em>Molecular Ecology Notes</em> 5 (1): 184–86. <a href="https://doi.org/10.1111/j.1471-8286.2004.00828.x">https://doi.org/10.1111/j.1471-8286.2004.00828.x</a>.
</div>
<div class="csl-entry">
Granato, Italo S. C., Giovanni Galli, Evellyn Giselly de Oliveira Couto, Massaine Bandeira e Souza, Leandro Freitas Mendonça, and Roberto Fritsche-Neto. 2018. <span>“snpReady: A Tool to Assist Breeders in Genomic Analysis.”</span> <em>Molecular Breeding</em> 38 (8). <a href="https://doi.org/10.1007/s11032-018-0844-8">https://doi.org/10.1007/s11032-018-0844-8</a>.
</div>
<div class="csl-entry">
Gu, Zuguang, Lei Gu, Roland Eils, Matthias Schlesner, and Benedikt Brors. 2014. <span>“<span><em>Circlize</em></span>implements and Enhances Circular Visualization in R.”</span> <em>Bioinformatics</em> 30 (19): 2811–12. <a href="https://doi.org/10.1093/bioinformatics/btu393">https://doi.org/10.1093/bioinformatics/btu393</a>.
</div>
<div class="csl-entry">
Jeong, Seongmun, Jae-Yoon Kim, Soon-Chun Jeong, Sung-Taeg Kang, Jung-Kyung Moon, and Namshin Kim. 2017. <span>“GenoCore: A Simple and Fast Algorithm for Core Subset Selection from Large Genotype Datasets.”</span> Edited by Alexandre G. de Brevern. <em>PLOS ONE</em> 12 (7): e0181420. <a href="https://doi.org/10.1371/journal.pone.0181420">https://doi.org/10.1371/journal.pone.0181420</a>.
</div>
<div class="csl-entry">
Jombart, Thibaut, Sébastien Devillard, and François Balloux. 2010. <span>“Discriminant Analysis of Principal Components: A New Method for the Analysis of Genetically Structured Populations.”</span> <em>BMC Genetics</em> 11 (1): 94. <a href="https://doi.org/10.1186/1471-2156-11-94">https://doi.org/10.1186/1471-2156-11-94</a>.
</div>
<div class="csl-entry">
Kamvar, Zhian N., Javier F. Tabima, and Niklaus J. Grünwald. 2014. <span>“<span><em>Poppr</em></span>: An R Package for Genetic Analysis of Populations with Clonal, Partially Clonal, and/or Sexual Reproduction.”</span> <em>PeerJ</em> 2 (March): e281. <a href="https://doi.org/10.7717/peerj.281">https://doi.org/10.7717/peerj.281</a>.
</div>
<div class="csl-entry">
Kang, Hyun Min, Jae Hoon Sul, Susan K Service, Noah A Zaitlen, Sit-yee Kong, Nelson B Freimer, Chiara Sabatti, and Eleazar Eskin. 2010. <span>“Variance Component Model to Account for Sample Structure in Genome-Wide Association Studies.”</span> <em>Nature Genetics</em> 42 (4): 348–54. <a href="https://doi.org/10.1038/ng.548">https://doi.org/10.1038/ng.548</a>.
</div>
<div class="csl-entry">
Luu, Keurcien, Eric Bazin, and Michael G. B. Blum. 2016. <span>“<span><em>Pcadapt</em></span>: An R Package to Perform Genome Scans for Selection Based on Principal Component Analysis.”</span> <em>Molecular Ecology Resources</em> 17 (1): 67–77. <a href="https://doi.org/10.1111/1755-0998.12592">https://doi.org/10.1111/1755-0998.12592</a>.
</div>
<div class="csl-entry">
Paradis, Emmanuel, and Klaus Schliep. 2018. <span>“Ape 5.0: An Environment for Modern Phylogenetics and Evolutionary Analyses in R.”</span> Edited by Russell Schwartz. <em>Bioinformatics</em> 35 (3): 526–28. <a href="https://doi.org/10.1093/bioinformatics/bty633">https://doi.org/10.1093/bioinformatics/bty633</a>.
</div>
<div class="csl-entry">
Whitlock, Michael C., and Katie E. Lotterhos. 2015. <span>“Reliable Detection of Loci Responsible for Local Adaptation: Inference of a Null Model Through Trimming the Distribution of<span><em>F</em></span><span><sub>ST</sub></span>.”</span> <em>The American Naturalist</em> 186 (S1): S24–36. <a href="https://doi.org/10.1086/682949">https://doi.org/10.1086/682949</a>.
</div>
<div class="csl-entry">
Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2016. <span>“Ggtree: An r Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.”</span> Edited by Greg McInerny. <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210x.12628">https://doi.org/10.1111/2041-210x.12628</a>.
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sec-ai-report.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/10-INDEX.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["ShiNyP-guide.pdf", "ShiNyP-guide.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
