<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>9 FAQ | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); Index AI Report 8 AMOVA (Analysis of MOlecular...">
<meta name="generator" content="bookdown 0.42 with bs4_book()">
<meta property="og:title" content="9 FAQ | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-index.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); Index AI Report 8 AMOVA (Analysis of MOlecular...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="9 FAQ | User Guide for ShiNyP">
<meta name="twitter:description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); Index AI Report 8 AMOVA (Analysis of MOlecular...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Lato-0.4.9/font.css" rel="stylesheet">
<link href="libs/Roboto_Mono-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="" href="sec-quickstart.html">Quickstart</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="active" href="sec-index.html"><span class="header-section-number">9</span> FAQ</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-index" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> FAQ<a class="anchor" aria-label="anchor" href="#sec-index"><i class="fas fa-link"></i></a>
</h1>
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-2662B66TL7"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2662B66TL7');
</script><div id="index" class="section level2 unnumbered">
<h2>Index<a class="anchor" aria-label="anchor" href="#index"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li><p>AI Report
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-ai-report.html#sec-ai-report">8</a></p></li>
<li><p>AMOVA (Analysis of MOlecular VAriance)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#amova-analysis-of-molecular-variance">5.4</a></p></li>
<li><p>API key
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-ai-report.html#step-2-ai-driven-report">8</a></p></li>
<li><p>Bayesian Information Criterion (BIC)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#step-2-dapc-analysis">4.2</a></p></li>
<li><p>Chromosome Info.
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-density">2.3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#circos-plot">5.2</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#ibs-identity-by-state">6.3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#manhattan-plot-plus">6.4</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-snp-set">7.2</a></p></li>
<li><p>Circos Plot
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#circos-plot">5.2</a></p></li>
<li><p>Core Sample Set
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-sample-set">7.1</a></p></li>
<li><p>Core SNP Set
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-snp-set">7.2</a></p></li>
<li><p>DAPC (Discriminant Analysis of Principal Components)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#dapc-discriminant-analysis-of-principal-components">4.2</a></p></li>
<li><p>data.frame
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-2-transform-to-data.frame">1.1</a></p></li>
<li><p>Demo Data
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-1-input-your-vcf-file">1.1</a></p></li>
<li><p>Diversity Parameter
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#diversity-parameter">5.1</a></p></li>
<li><p>Genetic Distance
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#genetic-distance">5.3</a></p></li>
<li><p>genind
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html#step-1-transform-data.frame-to-genind">3</a></p></li>
<li><p>genlight
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html#step-2-transform-genind-to-genlight">5</a></p></li>
<li><p>Group Info.
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html#step-1-transform-data.frame-to-genind">3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#kinship-analysis">4.5</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#scatter-plot-plus">4.6</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#diversity-parameter">5.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#outflank">6.2</a></p></li>
<li><p>Hardy-Weinberg equilibrium (HWE)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">2.2</a></p></li>
<li><p>Heterozygosity rate
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">2.1</a></p></li>
<li><p>IBS (Identity By State)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#ibs-identity-by-state">6.3</a></p></li>
<li><p>Kinship Analysis
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#kinship-analysis">4.5</a></p></li>
<li><p>Manhattan Plot
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#manhattan-plot-plus">6.4</a></p></li>
<li><p>Minor allele frequency (MAF)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">2.2</a></p></li>
<li><p>Missing rate
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">2.1</a></p></li>
<li><p>NJ (Neighbor-Joining) Tree
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#nj-neighbor-joining-tree">4.4</a></p></li>
<li><p>OutFLANK
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#outflank">6.2</a></p></li>
<li><p>PCA (Principal Component Analysis)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#pca-principal-component-analysis">4.1</a></p></li>
<li><p>pcadapt
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#pcadapt">6.1</a></p></li>
<li><p>Permutation Test
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#step-2-run-permutation-test">5.4</a></p></li>
<li><p>Sample QC
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">2.1</a></p></li>
<li><p>Scatter Plot
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">4.6</a></p></li>
<li><p><em>ShiNyP</em>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-shinyp.html#sec-shinyp">ShiNyP</a></p></li>
<li><p>Site Info.
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-2-transform-to-data.frame">1.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">2.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">2.2</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-density">2.3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#diversity-parameter">5.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#pcadapt">6.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#outflank">6.2</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#ibs-identity-by-state">6.3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-snp-set">7.2</a></p></li>
<li><p>SNP Density
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-density">2.3</a></p></li>
<li><p>SNP QC
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">2.2</a></p></li>
<li><p>Tree Plot
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#tree-plot-plus">4.7</a></p></li>
<li><p>UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree">4.3</a></p></li>
<li><p>VCF
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#vcf">1.1</a></p></li>
</ul>
<hr>
</div>
<div id="bibliography" class="section level2 unnumbered">
<h2>Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"><i class="fas fa-link"></i></a>
</h2>

<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-browning2021" class="csl-entry">
Browning, Brian L., Xiaowen Tian, Ying Zhou, and Sharon R. Browning. 2021. <span>“Fast Two-Stage Phasing of Large-Scale Sequence Data.”</span> <em>The American Journal of Human Genetics</em> 108 (10): 1880–90. <a href="https://doi.org/10.1016/j.ajhg.2021.08.005">https://doi.org/10.1016/j.ajhg.2021.08.005</a>.
</div>
<div id="ref-browning2018" class="csl-entry">
Browning, Brian L., Ying Zhou, and Sharon R. Browning. 2018. <span>“A One-Penny Imputed Genome from Next-Generation Reference Panels.”</span> <em>The American Journal of Human Genetics</em> 103 (3): 338–48. <a href="https://doi.org/10.1016/j.ajhg.2018.07.015">https://doi.org/10.1016/j.ajhg.2018.07.015</a>.
</div>
<div id="ref-excoffier1992" class="csl-entry">
Excoffier, L, P E Smouse, and J M Quattro. 1992. <span>“Analysis of Molecular Variance Inferred from Metric Distances Among DNA Haplotypes: Application to Human Mitochondrial DNA Restriction Data.”</span> <em>Genetics</em> 131 (2): 479–91. <a href="https://doi.org/10.1093/genetics/131.2.479">https://doi.org/10.1093/genetics/131.2.479</a>.
</div>
<div id="ref-goudet2004" class="csl-entry">
GOUDET, JÉRÔME. 2004. <span>“Hierfstat, a Package for r to Compute and Test Hierarchical <span> <em>F</em></span><span>-</span>Statistics.”</span> <em>Molecular Ecology Notes</em> 5 (1): 184–86. <a href="https://doi.org/10.1111/j.1471-8286.2004.00828.x">https://doi.org/10.1111/j.1471-8286.2004.00828.x</a>.
</div>
<div id="ref-granato2018" class="csl-entry">
Granato, Italo S. C., Giovanni Galli, Evellyn Giselly de Oliveira Couto, Massaine Bandeira e Souza, Leandro Freitas Mendonça, and Roberto Fritsche-Neto. 2018. <span>“snpReady: A Tool to Assist Breeders in Genomic Analysis.”</span> <em>Molecular Breeding</em> 38 (8). <a href="https://doi.org/10.1007/s11032-018-0844-8">https://doi.org/10.1007/s11032-018-0844-8</a>.
</div>
<div id="ref-gu2014" class="csl-entry">
Gu, Zuguang, Lei Gu, Roland Eils, Matthias Schlesner, and Benedikt Brors. 2014. <span>“<span><em>Circlize</em></span>implements and Enhances Circular Visualization in R.”</span> <em>Bioinformatics</em> 30 (19): 2811–12. <a href="https://doi.org/10.1093/bioinformatics/btu393">https://doi.org/10.1093/bioinformatics/btu393</a>.
</div>
<div id="ref-Jeong2017" class="csl-entry">
Jeong, Seongmun, Jae-Yoon Kim, Soon-Chun Jeong, Sung-Taeg Kang, Jung-Kyung Moon, and Namshin Kim. 2017. <span>“GenoCore: A Simple and Fast Algorithm for Core Subset Selection from Large Genotype Datasets.”</span> Edited by Alexandre G. de Brevern. <em>PLOS ONE</em> 12 (7): e0181420. <a href="https://doi.org/10.1371/journal.pone.0181420">https://doi.org/10.1371/journal.pone.0181420</a>.
</div>
<div id="ref-jombart2010" class="csl-entry">
Jombart, Thibaut, Sébastien Devillard, and François Balloux. 2010. <span>“Discriminant Analysis of Principal Components: A New Method for the Analysis of Genetically Structured Populations.”</span> <em>BMC Genetics</em> 11 (1): 94. <a href="https://doi.org/10.1186/1471-2156-11-94">https://doi.org/10.1186/1471-2156-11-94</a>.
</div>
<div id="ref-kamvar2014" class="csl-entry">
Kamvar, Zhian N., Javier F. Tabima, and Niklaus J. Grünwald. 2014. <span>“<span><em>Poppr</em></span>: An R Package for Genetic Analysis of Populations with Clonal, Partially Clonal, and/or Sexual Reproduction.”</span> <em>PeerJ</em> 2 (March): e281. <a href="https://doi.org/10.7717/peerj.281">https://doi.org/10.7717/peerj.281</a>.
</div>
<div id="ref-kang2010" class="csl-entry">
Kang, Hyun Min, Jae Hoon Sul, Susan K Service, Noah A Zaitlen, Sit-yee Kong, Nelson B Freimer, Chiara Sabatti, and Eleazar Eskin. 2010. <span>“Variance Component Model to Account for Sample Structure in Genome-Wide Association Studies.”</span> <em>Nature Genetics</em> 42 (4): 348–54. <a href="https://doi.org/10.1038/ng.548">https://doi.org/10.1038/ng.548</a>.
</div>
<div id="ref-Luu2017" class="csl-entry">
Luu, Keurcien, Eric Bazin, and Michael G. B. Blum. 2016. <span>“<span><em>Pcadapt</em></span>: An R Package to Perform Genome Scans for Selection Based on Principal Component Analysis.”</span> <em>Molecular Ecology Resources</em> 17 (1): 67–77. <a href="https://doi.org/10.1111/1755-0998.12592">https://doi.org/10.1111/1755-0998.12592</a>.
</div>
<div id="ref-paradis2018" class="csl-entry">
Paradis, Emmanuel, and Klaus Schliep. 2018. <span>“Ape 5.0: An Environment for Modern Phylogenetics and Evolutionary Analyses in R.”</span> Edited by Russell Schwartz. <em>Bioinformatics</em> 35 (3): 526–28. <a href="https://doi.org/10.1093/bioinformatics/bty633">https://doi.org/10.1093/bioinformatics/bty633</a>.
</div>
<div id="ref-whitlock2015" class="csl-entry">
Whitlock, Michael C., and Katie E. Lotterhos. 2015. <span>“Reliable Detection of Loci Responsible for Local Adaptation: Inference of a Null Model Through Trimming the Distribution of<span><em>F</em></span><span><sub>ST</sub></span>.”</span> <em>The American Naturalist</em> 186 (S1): S24–36. <a href="https://doi.org/10.1086/682949">https://doi.org/10.1086/682949</a>.
</div>
<div id="ref-yu2016" class="csl-entry">
Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2016. <span>“Ggtree: An r Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.”</span> Edited by Greg McInerny. <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210x.12628">https://doi.org/10.1111/2041-210x.12628</a>.
</div>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-index"><span class="header-section-number">9</span> FAQ</a></li>
<li><a class="nav-link" href="#index">Index</a></li>
<li><a class="nav-link" href="#bibliography">Bibliography</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/10-Index.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/10-Index.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on Apr 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
