<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Quickstart | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="Prerequisites Before installing ShiNyP, ensure your system meets the following requirements: R Version: R ≥ 4.4 (compatible with Bioconductor version 3.19) R Tools: Windows: Rtools macOS: Xcode...">
<meta name="generator" content="bookdown 0.40 with bs4_book()">
<meta property="og:title" content="Quickstart | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-quickstart.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="Prerequisites Before installing ShiNyP, ensure your system meets the following requirements: R Version: R ≥ 4.4 (compatible with Bioconductor version 3.19) R Tools: Windows: Rtools macOS: Xcode...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Quickstart | User Guide for ShiNyP">
<meta name="twitter:description" content="Prerequisites Before installing ShiNyP, ensure your system meets the following requirements: R Version: R ≥ 4.4 (compatible with Bioconductor version 3.19) R Tools: Windows: Rtools macOS: Xcode...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Lato-0.4.9/font.css" rel="stylesheet">
<link href="libs/Roboto_Mono-0.4.9/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.9/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="active" href="sec-quickstart.html">Quickstart</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="" href="sec-index.html"><span class="header-section-number">9</span> INDEX</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-quickstart" class="section level1 unnumbered">
<h1>Quickstart<a class="anchor" aria-label="anchor" href="#sec-quickstart"><i class="fas fa-link"></i></a>
</h1>
<div id="prerequisites" class="section level2 unnumbered">
<h2>Prerequisites<a class="anchor" aria-label="anchor" href="#prerequisites"><i class="fas fa-link"></i></a>
</h2>
<p>Before installing <span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span>, ensure your
system meets the following requirements:</p>
<ol style="list-style-type: decimal">
<li><p><strong>R Version</strong>: R ≥ 4.4 (compatible with Bioconductor version 3.19)</p></li>
<li>
<p><strong>R Tools</strong>:</p>
<ul>
<li><p><strong>Windows</strong>:
<a href="https://cran.r-project.org/bin/windows/Rtools/">Rtools</a></p></li>
<li><p><strong>macOS</strong>: <a href="https://developer.apple.com/xcode/">Xcode Command Line
Tools</a></p></li>
<li><p><strong>Linux</strong>: Development tools (e.g., <code>r-base-dev</code>)</p></li>
</ul>
</li>
</ol>
<hr>
<details><summary>
⚙️ <strong>Getting Started with R/RStudio/Rtools</strong>
</summary><p><strong>Step 1: Install R</strong></p>
<ol style="list-style-type: decimal">
<li>
<p>Visit the official R Project website: <a href="https://www.r-project.org/" class="uri">https://www.r-project.org/</a>.</p>
<ul>
<li><p><strong>Windows</strong>: Click <strong>Download R for Windows</strong>, select “base,”
and download the latest version.</p></li>
<li><p><strong>MacOS</strong>: Click <strong>Download R for macOS</strong> and choose the correct
version for your system.</p></li>
</ul>
</li>
<li><p>Install R by double-clicking the downloaded installer and following
the on-screen instructions. Use the default options unless specific
needs arise.</p></li>
</ol>
<p><strong>Step 2: Install RStudio</strong></p>
<ol style="list-style-type: decimal">
<li><p>Go to the RStudio official download page:
<a href="https://posit.co/download/rstudio-desktop/" class="uri">https://posit.co/download/rstudio-desktop/</a>.</p></li>
<li><p>Click <strong>Download</strong> under “RStudio Desktop - Open Source License.”</p></li>
<li>
<p>Select the version suitable for your operating system:</p>
<ul>
<li><p><strong>Windows</strong>: <code>.exe</code> file.</p></li>
<li><p><strong>MacOS</strong>: <code>.dmg</code> file.</p></li>
</ul>
</li>
<li>
<p>Locate the downloaded file and double-click it to start the
installation.</p>
<ul>
<li><p><strong>Windows</strong>: Run the <code>.exe</code> installer and follow the wizard
steps.</p></li>
<li><p><strong>MacOS</strong>: Drag the RStudio icon into the Applications folder.</p></li>
</ul>
</li>
<li><p>After installation, launch RStudio.</p></li>
</ol>
<p><strong>Step 3: Launch RStudio</strong></p>
<ol style="list-style-type: decimal">
<li><p>Open RStudio by clicking the shortcut created during installation or
searching for “RStudio” in your system’s application launcher.</p></li>
<li><p>RStudio will automatically detect your R installation and link to
it.</p></li>
<li>
<p>RStudio has four main panels:</p>
<p><strong>Console (Bottom-left)</strong>: Where you run R commands.</p>
<p><strong>Source (Top-left)</strong>: For writing and editing scripts.</p>
<p><strong>Environment/History (Top-right)</strong>: Displays objects, variables,
and command history.</p>
<p><strong>Plots/Files/Help (Bottom-right)</strong>: Displays plots, files, and R
documentation.</p>
</li>
</ol>
<p><strong>Step 5: Install Rtools</strong></p>
<ul>
<li>
<p><strong>Windows</strong>: Download and install the newest 64-bit
<a href="https://cran.r-project.org/bin/windows/Rtools/"><strong>Rtools</strong></a>installer
from CRAN. Rtools is a toolchain bundle required for building R
packages from source, particularly those that require C/C++ or
Fortran code compilation.</p>
<p>Run the file to install <strong>Rtools</strong>, following the onscreen
instructions. If unsure, keep the default settings. Then, return to
this page and continue.</p>
<p>First, run <code>Sys.which("make")</code>. If it detects <strong>make.exe</strong>, you’re
all set. If not, configure the system path.</p>
</li>
<li>
<p><strong>MacOS</strong>: Open Terminal: Press <code>Cmd + Space</code>, type “Terminal”, and
press <code>Enter</code>. Run the following command and press <code>Enter</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">xcode</span><span class="op">-</span><span class="va">select</span> <span class="op">-</span><span class="op">-</span><span class="va">install</span></span></code></pre></div>
<p>Enter your password if prompted and follow the onscreen
instructions.</p>
<p>Verify Installation:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">xcode</span><span class="op">-</span><span class="va">select</span> <span class="op">-</span><span class="va">p</span></span></code></pre></div>
<p>It should return <code>/Library/Developer/CommandLineTools</code>.</p>
</li>
</ul></details><hr>
<blockquote>
<p><strong>Note</strong>: <span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span> is currently in the
Beta version. Ensure you are comfortable working with beta software
and reporting any issues you encounter. For any inquiries, please
email us at:
<a href="mailto:teddyhuangyh@gmail.com" class="email">teddyhuangyh@gmail.com</a>.</p>
</blockquote>
<hr>
</div>
<div id="installing-shinyp" class="section level2 unnumbered">
<h2>Installing ShiNyP<a class="anchor" aria-label="anchor" href="#installing-shinyp"><i class="fas fa-link"></i></a>
</h2>
<p><strong>Step 1: Pre-install Required Package</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html">install</a></span><span class="op">(</span>version <span class="op">=</span> <span class="st">"3.19"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"qvalue"</span>, <span class="st">"SNPRelate"</span>, <span class="st">"ggtree"</span><span class="op">)</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><strong>Step 2: Install the <span style="color: #9D6853;"><em>ShiNyP</em></span></strong> <strong>Package
from GitHub</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span> </span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"TeddYenn/ShiNyP"</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><strong>Step 3: Start the <span style="color: #9D6853;"><em>ShiNyP</em></span></strong> <strong>Platform</strong></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ShiNyP</span><span class="op">)</span></span>
<span><span class="fu">ShiNyP</span><span class="fu">::</span><span class="fu">run_ShiNyP</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><strong>Step 4: Run <span style="color: #9D6853;"><em>ShiNyP</em></span></strong> <strong>Analysis</strong></p>
<p>Input your SNP data in VCF format, or feel free to use our <span style="color: #1e90ff;"><strong>Demo
Data</strong></span>.</p>
<blockquote>
<p><strong>Note:</strong> If you run in <u>RStudio</u>, you can click the
<span style="color: #1e90ff;"><strong>Open in Browser</strong></span> button.</p>
</blockquote>
<hr>
</div>
<div id="troubleshooting" class="section level2 unnumbered">
<h2>Troubleshooting<a class="anchor" aria-label="anchor" href="#troubleshooting"><i class="fas fa-link"></i></a>
</h2>
<p>During installation or while running
<span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span>, you might encounter issues.
Below are common problems and their solutions.</p>
<dl>
<dt>⚒️ <strong>R and Bioconductor Version Mismatch</strong>
</dt>
<dd>
<p>The installation specifies Bioconductor version 3.19, which requires
R version ≥ 4.4. If your R version is below 4.4, update it from
CRAN.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check your R version</span></span>
<span><span class="va">R.version.string</span></span></code></pre></div>
</dd>
</dl>
<hr>
<dl>
<dt>⚒️ <strong>Installation of Dependencies</strong>
</dt>
<dd>
<p>Installing packages like <em>shiny</em> and <em>dartR</em> may require
<strong>additional developer tools</strong>. You might encounter error messages
such as:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="sec-quickstart.html#cb7-1" tabindex="-1"></a>ERROR<span class="sc">:</span> dependencies <span class="st">'shiny'</span>, <span class="st">'dartR'</span> are not available <span class="cf">for</span> package <span class="st">'ShiNyP'</span></span></code></pre></div>
</dd>
<dd>
<ul>
<li>
<p><strong>For Windows</strong>:</p>
<p>Download Rtools from <a href="https://cran.r-project.org/bin/windows/Rtools/"><strong>CRAN
Rtools</strong></a>.</p>
<p>Run the installer and follow the on-screen instructions.</p>
<p>Ensure Rtools is added to your system PATH during installation.</p>
<p>Verify Installation:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">pkgbuild</span><span class="fu">::</span><span class="fu"><a href="https://pkgbuild.r-lib.org/reference/has_rtools.html">find_rtools</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>It should return <code>TRUE</code> if Rtools is correctly installed.</p>
</li>
<li>
<p><strong>For macOS</strong>:</p>
<p>Open Terminal: Press <code>Cmd + Space</code>, type “Terminal”, and press
<code>Enter</code>.</p>
<p>Run the Installation Command:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">xcode</span><span class="op">-</span><span class="va">select</span> <span class="op">-</span><span class="op">-</span><span class="va">install</span></span></code></pre></div>
<p>Verify Installation:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">xcode</span><span class="op">-</span><span class="va">select</span> <span class="op">-</span><span class="va">p</span></span></code></pre></div>
<p>It should return <code>/Library/Developer/CommandLineTools</code>.</p>
</li>
<li>
<p><strong>For Linux</strong>:</p>
<p>Install Essential Build Tools:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="sec-quickstart.html#cb11-1" tabindex="-1"></a>sudo apt<span class="sc">-</span>get install r<span class="sc">-</span>base<span class="sc">-</span>dev</span>
<span id="cb11-2"><a href="sec-quickstart.html#cb11-2" tabindex="-1"></a>sudo apt<span class="sc">-</span>get install build<span class="sc">-</span>essential libcurl4<span class="sc">-</span>gnutls<span class="sc">-</span>dev libxml2<span class="sc">-</span>dev libssl<span class="sc">-</span>dev</span></code></pre></div>
</li>
</ul>
</dd>
</dl>
<hr>
<dl>
<dt>⚒️ <strong>GitHub Installation Issues</strong>
</dt>
<dd>
<p>Verify that your system is connected to the internet. Check for any
firewall or proxy settings that might block GitHub access.</p>
</dd>
</dl>
<hr>
<dl>
<dt>⚒️ <strong>Package Loading Issues</strong>
</dt>
<dd>
<p>Errors occur when loading <span style="color: #9D6853;"><strong><em>ShiNyP</em></strong></span> or
its dependencies, possibly due to outdated packages or conflicts.</p>
<ol style="list-style-type: decimal">
<li>
<p><strong>Update All Installed Packages</strong>:</p>
<p>When prompted with</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># These packages have more recent versions available.</span></span>
<span><span class="co"># It is recommended to update all of them.</span></span>
<span><span class="co"># Which would you like to update?</span></span>
<span></span>
<span><span class="co"># -&gt; Respond with 1 to update all packages.</span></span></code></pre></div>
<p><strong>Please type 1 (1: All)</strong></p>
</li>
<li>
<p>If a specific package is causing issues, reinstall it.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"dartRverse"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html">install</a></span><span class="op">(</span><span class="st">"SNPRelate"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"irudnyts/openai"</span>, ref <span class="op">=</span> <span class="st">"r6"</span>, force <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</li>
<li>
<p>Check for Namespace Conflicts:</p>
<p>Identify if multiple packages are masking functions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/conflicts.html">conflicts</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</li>
</ol>
</dd>
</dl>
<hr>
</div>
<div id="main-functions" class="section level2 unnumbered">
<h2>Main Functions<a class="anchor" aria-label="anchor" href="#main-functions"><i class="fas fa-link"></i></a>
</h2>
<div class="inline-figure"><img src="images/Fig.%201-02-01.jpg" width="650"></div>
<div class="line-block">▲ <strong>An overview of the <span style="color: #9D6853;"><em>ShiNyP</em></span> platform’s workflow for genome-wide SNP data analysis.</strong><br>
◾ <strong>Data Input &amp; Processing</strong>: Beginning with <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html" target="_blank">VCF data input</a>, it performs <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html" target="_blank">quality control (QC)</a> and <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html" target="_blank">data transformation</a> steps.<br>
◾ <strong>Analysis &amp; Output</strong>: The analysis phase is divided into modules, each represented as a page in the platform, with multiple subpages offering specific analytical functions, including <a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html" target="_blank">population structure</a>, <a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html" target="_blank">genetic diversity</a>, <a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html" target="_blank">selection sweep</a>, and <a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html" target="_blank">core collection</a>.<br>
◾ <strong>Customized Output</strong>: The final output offers tailored visualizations and includes an AI-generated report summarizing the results. The pipeline streamlines data input, processing, and advanced analysis to deliver publication-ready figures and reports customized to the user’s needs.<br>
*Subpage frame colors indicate available functions for customization. For example, blue frames for <u>PCA</u> and <u>DAPC</u> correspond to the Scatter Plot <sup>Plus</sup> tool for customizing scatter plots, while red and purple frames correspond to Tree Plot <sup>Plus</sup> and Manhattan Plot <sup>Plus</sup>, respectively.</div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="index.html">Welcome to ShiNyP</a></div>
<div class="next"><a href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-quickstart">Quickstart</a></li>
<li><a class="nav-link" href="#prerequisites">Prerequisites</a></li>
<li><a class="nav-link" href="#installing-shinyp">Installing ShiNyP</a></li>
<li><a class="nav-link" href="#troubleshooting">Troubleshooting</a></li>
<li><a class="nav-link" href="#main-functions">Main Functions</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/index.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/index.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on Feb 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
