<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>3 Data Transform | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); ‚û°Ô∏è This section allow you to convert your SNP data...">
<meta name="generator" content="bookdown 0.42 with bs4_book()">
<meta property="og:title" content="3 Data Transform | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-data-conversion.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); ‚û°Ô∏è This section allow you to convert your SNP data...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="3 Data Transform | User Guide for ShiNyP">
<meta name="twitter:description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); ‚û°Ô∏è This section allow you to convert your SNP data...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Noto_Sans-0.4.10/font.css" rel="stylesheet">
<link href="libs/Ubuntu_Sans-0.4.10/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.10/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link rel="shortcut icon" href="favicon/favicon.ico">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="" href="get-started.html">Get Started üöÄ</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="active" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="" href="sec-index.html"><span class="header-section-number">9</span> FAQ</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-data-conversion" class="section level1" number="3">
<h1>
<span class="header-section-number">3</span> Data Transform<a class="anchor" aria-label="anchor" href="#sec-data-conversion"><i class="fas fa-link"></i></a>
</h1>
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-2662B66TL7"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2662B66TL7');
</script><div style="background-color: #7c6f41; color: white; padding: 10px 5px; margin: 10px 0px; border-radius: 15px;">
<p>‚û°Ô∏è This section allow you to convert your SNP data in <em>data.frame</em> into
multiple formats, including <em>genind</em>, <em>genind</em> and PLINK.</p>
</div>
<div id="required-file-4" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Required File:<a class="anchor" aria-label="anchor" href="#required-file-4"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>Input VCF Data (<em>data.frame</em> file) from the <u>Data Input</u>
page or</li>
<li>Post-QC Data (<em>data.frame</em> file) from the <u>Data QC</u>
page.</li>
</ul>
<hr>
</div>
<div id="step-1-transform-data.frame-to-genlight" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 1: Transform <em>data.frame</em> to <em>genlight</em><a class="anchor" aria-label="anchor" href="#step-1-transform-data.frame-to-genlight"><i class="fas fa-link"></i></a>
</h4>
<ol style="list-style-type: decimal">
<li>Click <span style="color: #fff;background-color: #007ACC;">Transform to
genlight</span>.</li>
<li>Download the generated <em>genlight</em> (in RDS format) to skip VCF or
<em>data.frame</em> upload next time by directly importing this file.</li>
</ol>
</div>
<div id="output" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Output:<a class="anchor" aria-label="anchor" href="#output"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<em>genlight</em> (RDS): <em>genlight</em> file ‚Äî required for downstream
analysis.</li>
</ul>
<hr>
<p><br></p>
</div>
<div id="step-2-transform-genlight-to-others" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Step 2: Transform <em>genlight</em> to others<a class="anchor" aria-label="anchor" href="#step-2-transform-genlight-to-others"><i class="fas fa-link"></i></a>
</h4>
<p>Select the desired data format to export from <em>genlight</em> and click
<span style="color: #fff;background-color: #007ACC;">Transform</span>.</p>
</div>
<div id="outputs-4" class="section level4 unnumbered" style="padding: 0px 5px; margin: 0px 10px">
<h4>Outputs:<a class="anchor" aria-label="anchor" href="#outputs-4"><i class="fas fa-link"></i></a>
</h4>
<ul>
<li>
<em>genlight</em> (RDS): <em>genlight</em> file with Group Info. ‚Äî required for
downstream analysis.</li>
<li>
<em>genind</em> (RDS): One of the input format for <em>ShiNyP</em>
<u>DAPC</u> subpage, optimized for DAPC analysis to reduce
computation time.</li>
</ul>
<p>The following transformed files will be generated at the specified path
you provide.</p>
<ul>
<li>PLINK (PED &amp; MAP): Input format for
<a href="https://www.cog-genomics.org/plink/1.9/input#ped">PLINK</a> program,
designed to perform a range of basic and large-scale SNP analyses.</li>
<li>GenAlEx (CSV): Input format for
<a href="https://biology-assets.anu.edu.au/GenAlEx/Welcome.html">GenAlEx</a>
program, offers a wide range of population genetic analysis in
Excel.</li>
<li>LEA (GENO &amp; LFMM): Input format for
<a href="https://www.bioconductor.org/packages/release/bioc/html/LEA.html">LEA</a>
R package, designed for population genomics, landscape genomics and
genotype-environment association tests.</li>
<li>GDS (GDS): Input format for
<a href="https://bioconductor.org/packages/release/bioc/html/SNPRelate.html">SNPRelate</a>
R package, designed for efficient SNP data analysis.</li>
<li>STRUCTURE (STR): Input format for
<a href="https://web.stanford.edu/group/pritchardlab/structure.html">STRUCTURE</a>
program, used for inferring population structure.</li>
<li>fastStructure (STR): Input format for
<a href="https://rajanil.github.io/fastStructure/">fastStructure</a> program,
used for inferring population structure from large SNP data.</li>
<li>PHYLIP (TXT): Input format for
<a href="https://phylipweb.github.io/phylip/">PHYLIP</a> program, used for
phylogenetic tree reconstruction and evolutionary analysis.</li>
<li>Treemix (GZ): Input format for
<a href="https://speciationgenomics.github.io/Treemix/">Treemix</a> program,
designed for modeling population splits and migration events.</li>
<li>BayeScan (TXT): Input format for
<a href="https://cmpg.unibe.ch/software/BayeScan/files/BayeScan2.0_manual.pdf">BayeScan</a>,
used for detecting loci under selection.</li>
</ul>
<div class="inline-figure"><img src="screenshots/DataTransform1.png" width="650"></div>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></div>
<div class="next"><a href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#sec-data-conversion"><span class="header-section-number">3</span> Data Transform</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/04-DataConversation.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/04-DataConversation.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on Apr 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
