<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>9 FAQ | User Guide for ShiNyP</title>
<meta name="author" content="Yen-Hsiang Huang, Chung-Feng Kao">
<meta name="description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); If you encounter any issues or have suggestions...">
<meta name="generator" content="bookdown 0.42 with bs4_book()">
<meta property="og:title" content="9 FAQ | User Guide for ShiNyP">
<meta property="og:type" content="book">
<meta property="og:url" content="https://github.com/TeddYenn/ShiNyP-guide/sec-index.html">
<meta property="og:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<meta property="og:description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); If you encounter any issues or have suggestions...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="9 FAQ | User Guide for ShiNyP">
<meta name="twitter:description" content="window.dataLayer = window.dataLayer || [];  function gtag(){dataLayer.push(arguments);}  gtag('js', new Date());  gtag('config', 'G-2662B66TL7'); If you encounter any issues or have suggestions...">
<meta name="twitter:image" content="https://github.com/TeddYenn/ShiNyP-guide/images/cover.png">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><link href="libs/Noto_Sans-0.4.10/font.css" rel="stylesheet">
<link href="libs/Ubuntu_Sans-0.4.10/font.css" rel="stylesheet">
<link href="libs/Montserrat-0.4.10/font.css" rel="stylesheet">
<script src="libs/bs3compat-0.9.0/transition.js"></script><script src="libs/bs3compat-0.9.0/tabs.js"></script><script src="libs/bs3compat-0.9.0/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><link rel="shortcut icon" href="favicon/favicon.ico">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
<style type="text/css">
    /* Used with Pandoc 2.11+ new --citeproc when CSL is used */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
          margin-bottom: 0em;
        }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="SNP Analysis and Visualization Platform">User Guide for ShiNyP</a>:
        <small class="text-muted">SNP Analysis and Visualization Platform</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Welcome to ShiNyP</a></li>
<li><a class="" href="get-started.html">Get Started 🚀</a></li>
<li><a class="" href="sec-data-input.html"><span class="header-section-number">1</span> Data Input</a></li>
<li><a class="" href="sec-data-qc.html"><span class="header-section-number">2</span> Data QC</a></li>
<li><a class="" href="sec-data-conversion.html"><span class="header-section-number">3</span> Data Transform</a></li>
<li><a class="" href="sec-population-structure.html"><span class="header-section-number">4</span> Population Structure</a></li>
<li><a class="" href="sec-genetic-diversity.html"><span class="header-section-number">5</span> Genetic Diversity</a></li>
<li><a class="" href="sec-selection-sweep.html"><span class="header-section-number">6</span> Selection Sweep</a></li>
<li><a class="" href="sec-core-collection.html"><span class="header-section-number">7</span> Core Collection</a></li>
<li><a class="" href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></li>
<li><a class="active" href="sec-index.html"><span class="header-section-number">9</span> FAQ</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/TeddYenn/ShiNyP-guide">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="sec-index" class="section level1" number="9">
<h1>
<span class="header-section-number">9</span> FAQ<a class="anchor" aria-label="anchor" href="#sec-index"><i class="fas fa-link"></i></a>
</h1>
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-2662B66TL7"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2662B66TL7');
</script><blockquote>
<p><strong>If you encounter any issues or have suggestions for new features,
please submit a request on the <a href="https://github.com/TeddYenn/ShiNyP/issues">GitHub Issues
page</a> or email us
at: <a href="mailto:teddyhuangyh@gmail.com">teddyhuangyh@gmail.com</a></strong></p>
</blockquote>
<div id="general-usage" class="section level2 unnumbered">
<h2>General Usage<a class="anchor" aria-label="anchor" href="#general-usage"><i class="fas fa-link"></i></a>
</h2>
<p>docker run -d -p 3838:3838 teddyenn/shinyp-platfor</p>
<ol style="list-style-type: decimal">
<li>
<p><strong>Can I use the online version of <em>ShiNyP</em> platform?</strong></p>
<p>Yes, a trial version of <em>ShiNyP</em> is available online at:
<a href="https://teddyhuang.shinyapps.io/ShiNyP_Demo/" class="uri">https://teddyhuang.shinyapps.io/ShiNyP_Demo/</a></p>
<p>This web-based DEMO is hosted on Shinyapps.io and is intended for
trial purposes. Please note that, due to server memory limitations
(1GB RAM), this version is not suitable for the analysis of
large-scale SNP datasets. For complete functionality and to analyze
larger datasets, we strongly recommend downloading <em>ShiNyP</em> from
GitHub.</p>
<p>For more information: <a href="https://teddyenn.github.io/ShiNyP-guide/get-started.html">Get
Started</a></p>
</li>
<li>
<p><strong>Do I need programming skills to use <em>ShiNyP</em>?</strong></p>
<p>No programming experience is required. <em>ShiNyP</em> provides a
user-friendly graphical interface that allows you to perform all
analyses interactively without coding.</p>
</li>
<li>
<p><strong>How do I install and start the <em>ShiNyP</em> platform?</strong></p>
<p>You can install <em>ShiNyP</em> in R via:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"TeddYenn/ShiNyP"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ShiNyP</span><span class="op">)</span></span>
<span><span class="fu">ShiNyP</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ShiNyP/man/run_ShiNyP.html">run_ShiNyP</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Alternatively, use the Docker image for a one-step deployment:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="sec-index.html#cb6-1" tabindex="-1"></a><span class="ex">docker</span> run <span class="at">-d</span> <span class="at">-p</span> 3838:3838 teddyenn/shinyp-platfor</span></code></pre></div>
<p>For more information: <a href="https://teddyenn.github.io/ShiNyP-guide/get-started.html">Get
Started</a></p>
</li>
<li>
<p><strong>Is my data secure when using <em>ShiNyP</em>?</strong></p>
<p>All analyses are conducted locally on your machine or server. No
user data is transmitted to third-party servers unless you
explicitly use AI features (<em>ShiNyP</em> AI chatbot and reporting
system).</p>
</li>
<li>
<p><strong>Does <em>ShiNyP</em> require internet access to function?</strong></p>
<p><em>ShiNyP</em> runs locally and does not require internet access for core
analyses. Internet is only needed when installing the packages and
using AI features (<em>ShiNyP</em> AI chatbot and reporting system).</p>
</li>
</ol>
<hr>
</div>
<div id="analysis-features" class="section level2 unnumbered">
<h2>Analysis &amp; Features<a class="anchor" aria-label="anchor" href="#analysis-features"><i class="fas fa-link"></i></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<p><strong>What types of input data are supported by <em>ShiNyP</em>?</strong></p>
<p><em>ShiNyP</em> supports genome-wide biallelic SNP datasets in Variant Call
Format (VCF). It is also compatible with <code>data.frame</code> and <code>genlight</code>
files, covering both diploid and polyploid species.</p>
<p>For more information: <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html">Data
Input</a></p>
</li>
<li>
<p><strong>What kind of output does the platform provide?</strong></p>
<p>The platform generates publication-ready figures (PDF, PNG, or JPEG)
and tables, reusable R data objects, and AI-assisted reports that
summarize analytical results in natural language.</p>
</li>
<li>
<p><strong>Can I customize analysis parameters?</strong></p>
<p>Yes, users can adjust analysis thresholds, models, and filtering
criteria directly through the interface to suit their research
needs.</p>
</li>
<li>
<p><strong>Can I analyze large SNP datasets with <em>ShiNyP</em>?</strong></p>
<p>Yes, <em>ShiNyP</em> is optimized for both moderate and large-scale SNP
datasets. However, performance may depend on your system’s hardware
specifications.</p>
</li>
<li>
<p><strong>Can I analyze datasets from different species in one session?</strong></p>
<p>Yes, as long as your data are properly formatted, <em>ShiNyP</em> supports
datasets from different species. However, it is recommended to
analyze each species separately for clearer results.</p>
</li>
<li>
<p><strong>Can <em>ShiNyP</em> export data to formats used by other genetics
software?</strong></p>
<p>Yes, results and transformed data can be exported in formats
compatible with tools such as STRUCTURE, PLINK, GenAlEx, and others.</p>
<p>For more information: <a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html">Data
Transform</a></p>
</li>
</ol>
<hr>
</div>
<div id="support" class="section level2 unnumbered">
<h2>Support<a class="anchor" aria-label="anchor" href="#support"><i class="fas fa-link"></i></a>
</h2>
<ol style="list-style-type: decimal">
<li>
<p><strong>Is <em>ShiNyP</em> open-source? Is it possible to extend ShiNyP with
custom modules or scripts?</strong></p>
<p>Yes, <em>ShiNyP</em> is released under an open-source license (<a href="https://www.gnu.org/licenses/gpl-3.0.html.en">GNU General
Public License</a>). The
source code is available on GitHub for transparency and community
contributions.</p>
</li>
<li>
<p><strong>Where can I report bugs or request new features?</strong></p>
<p>If you encounter any issues or have suggestions for new features,
please submit a request on the <a href="https://github.com/TeddYenn/ShiNyP/issues">GitHub Issues
page</a> or email us
at: <a href="#id_0" class="email">teddyhuangyh@gmail.com</a>.</p>
</li>
</ol>
<hr>
</div>
<div id="installation-issues" class="section level2 unnumbered">
<h2>Instal<strong>lation Issues</strong><a class="anchor" aria-label="anchor" href="#installation-issues"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>
<p><strong>R and Bioconductor version mismatch</strong></p>
<p>The installation specifies Bioconductor version 3.21, which requires
R version ≥ 4.5. If your R version is below 4.5, update it from
<a href="https://cran.r-project.org/">CRAN</a>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check your R version </span></span>
<span><span class="va">R.version.string</span></span></code></pre></div>
</li>
<li><p>xcode-select –install</p></li>
<li>
<p><strong>Installation of dependencies</strong></p>
<p>Installing packages like <em>shiny</em> and <em>dartR</em> may require additional
developer tools. You might encounter error messages such as:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="sec-index.html#cb8-1" tabindex="-1"></a>ERROR<span class="sc">:</span> dependencies <span class="st">'shiny'</span>, <span class="st">'dartR'</span> are not available <span class="cf">for</span> package <span class="st">'ShiNyP'</span></span></code></pre></div>
<ul>
<li><p><strong>For Windows</strong>: Download Rtools from <a href="https://cran.r-project.org/bin/windows/Rtools/"><strong>CRAN
Rtools</strong></a>.</p></li>
<li>
<p><strong>For macOS</strong>: Open Terminal and run the installation command:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="sec-index.html#cb9-1" tabindex="-1"></a><span class="ex">xcode-select</span> <span class="at">--install</span></span></code></pre></div>
</li>
</ul>
</li>
<li>
<p><strong>GitHub installation issues</strong></p>
<p>Verify that your system is connected to the internet. Check for any
firewall or proxy settings that might block GitHub access.</p>
</li>
<li>
<p><strong>Package loading issues</strong></p>
<p>Errors occur when loading <em>ShiNyP</em> or its dependencies, possibly due
to outdated packages or conflicts. <strong>Update all installed
packages.</strong></p>
</li>
</ul>
<hr>
</div>
<div id="index" class="section level2 unnumbered">
<h2>Index<a class="anchor" aria-label="anchor" href="#index"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>AI Report
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-ai-report.html#sec-ai-report">8</a>
</li>
<li>AMOVA (Analysis of MOlecular VAriance)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#amova-analysis-of-molecular-variance">5.4</a>
</li>
<li>API key
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-ai-report.html#step-2-ai-driven-report">8</a>
</li>
<li>Bayesian Information Criterion (BIC)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#step-2-dapc-analysis">4.2</a>
</li>
<li>Chromosome Info.
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-density">2.3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#circos-plot">5.2</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#ibs-identity-by-state">6.3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#manhattan-plot-plus">6.4</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-snp-set">7.2</a>
</li>
<li>Circos Plot
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#circos-plot">5.2</a>
</li>
<li>Core Sample Set
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-sample-set">7.1</a>
</li>
<li>Core SNP Set
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-snp-set">7.2</a>
</li>
<li>DAPC (Discriminant Analysis of Principal Components)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#dapc-discriminant-analysis-of-principal-components">4.2</a>
</li>
<li>data.frame
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-2-transform-to-data.frame">1.1</a>
</li>
<li>Demo Data
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-1-input-your-vcf-file">1.1</a>
</li>
<li>Diversity Parameter
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#diversity-parameter">5.1</a>
</li>
<li>Genetic Distance
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#genetic-distance">5.3</a>
</li>
<li>genind
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html#step-1-transform-data.frame-to-genind">3</a>
</li>
<li>genlight
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html#step-2-transform-genind-to-genlight">5</a>
</li>
<li>Group Info.
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-conversion.html#step-1-transform-data.frame-to-genind">3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#kinship-analysis">4.5</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#scatter-plot-plus">4.6</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#diversity-parameter">5.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#outflank">6.2</a>
</li>
<li>Hardy-Weinberg equilibrium (HWE)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">2.2</a>
</li>
<li>Heterozygosity rate
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">2.1</a>
</li>
<li>IBS (Identity By State)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#ibs-identity-by-state">6.3</a>
</li>
<li>Kinship Analysis
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#kinship-analysis">4.5</a>
</li>
<li>Manhattan Plot
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#manhattan-plot-plus">6.4</a>
</li>
<li>Minor allele frequency (MAF)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">2.2</a>
</li>
<li>Missing rate
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">2.1</a>
</li>
<li>NJ (Neighbor-Joining) Tree
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#nj-neighbor-joining-tree">4.4</a>
</li>
<li>OutFLANK
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#outflank">6.2</a>
</li>
<li>PCA (Principal Component Analysis)
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#pca-principal-component-analysis">4.1</a>
</li>
<li>pcadapt
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#pcadapt">6.1</a>
</li>
<li>Permutation Test
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#step-2-run-permutation-test">5.4</a>
</li>
<li>Sample QC
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">2.1</a>
</li>
<li>Scatter Plot
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">4.6</a>
</li>
<li>
<em>ShiNyP</em>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-shinyp.html#sec-shinyp">ShiNyP</a>
</li>
<li>Site Info.
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#step-2-transform-to-data.frame">1.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#sample-qc">2.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">2.2</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-density">2.3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-genetic-diversity.html#diversity-parameter">5.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#pcadapt">6.1</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#outflank">6.2</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-selection-sweep.html#ibs-identity-by-state">6.3</a>
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-core-collection.html#core-snp-set">7.2</a>
</li>
<li>SNP Density
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-density">2.3</a>
</li>
<li>SNP QC
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-qc.html#snp-qc">2.2</a>
</li>
<li>Tree Plot
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#tree-plot-plus">4.7</a>
</li>
<li>UPGMA (Unweighted Pair Group Method with Arithmetic mean) Tree
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-population-structure.html#upgma-unweighted-pair-group-method-with-arithmetic-mean-tree">4.3</a>
</li>
<li>VCF
<a href="https://teddyenn.github.io/ShiNyP-guide/sec-data-input.html#vcf">1.1</a>
</li>
</ul>
<hr>
</div>
<div id="bibliography" class="section level2 unnumbered">
<h2>Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"><i class="fas fa-link"></i></a>
</h2>

<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-excoffier1992" class="csl-entry">
Excoffier, L, P E Smouse, and J M Quattro. 1992. <span>“Analysis of Molecular Variance Inferred from Metric Distances Among DNA Haplotypes: Application to Human Mitochondrial DNA Restriction Data.”</span> <em>Genetics</em> 131 (2): 479–91. <a href="https://doi.org/10.1093/genetics/131.2.479">https://doi.org/10.1093/genetics/131.2.479</a>.
</div>
<div id="ref-goudet2004" class="csl-entry">
GOUDET, JÉRÔME. 2004. <span>“Hierfstat, a Package for r to Compute and Test Hierarchical <span> <em>F</em></span><span>-</span>Statistics.”</span> <em>Molecular Ecology Notes</em> 5 (1): 184–86. <a href="https://doi.org/10.1111/j.1471-8286.2004.00828.x">https://doi.org/10.1111/j.1471-8286.2004.00828.x</a>.
</div>
<div id="ref-granato2018" class="csl-entry">
Granato, Italo S. C., Giovanni Galli, Evellyn Giselly de Oliveira Couto, Massaine Bandeira e Souza, Leandro Freitas Mendonça, and Roberto Fritsche-Neto. 2018. <span>“snpReady: A Tool to Assist Breeders in Genomic Analysis.”</span> <em>Molecular Breeding</em> 38 (8). <a href="https://doi.org/10.1007/s11032-018-0844-8">https://doi.org/10.1007/s11032-018-0844-8</a>.
</div>
<div id="ref-gu2014" class="csl-entry">
Gu, Zuguang, Lei Gu, Roland Eils, Matthias Schlesner, and Benedikt Brors. 2014. <span>“<span><em>Circlize</em></span>implements and Enhances Circular Visualization in R.”</span> <em>Bioinformatics</em> 30 (19): 2811–12. <a href="https://doi.org/10.1093/bioinformatics/btu393">https://doi.org/10.1093/bioinformatics/btu393</a>.
</div>
<div id="ref-Jeong2017" class="csl-entry">
Jeong, Seongmun, Jae-Yoon Kim, Soon-Chun Jeong, Sung-Taeg Kang, Jung-Kyung Moon, and Namshin Kim. 2017. <span>“GenoCore: A Simple and Fast Algorithm for Core Subset Selection from Large Genotype Datasets.”</span> Edited by Alexandre G. de Brevern. <em>PLOS ONE</em> 12 (7): e0181420. <a href="https://doi.org/10.1371/journal.pone.0181420">https://doi.org/10.1371/journal.pone.0181420</a>.
</div>
<div id="ref-jombart2010" class="csl-entry">
Jombart, Thibaut, Sébastien Devillard, and François Balloux. 2010. <span>“Discriminant Analysis of Principal Components: A New Method for the Analysis of Genetically Structured Populations.”</span> <em>BMC Genetics</em> 11 (1): 94. <a href="https://doi.org/10.1186/1471-2156-11-94">https://doi.org/10.1186/1471-2156-11-94</a>.
</div>
<div id="ref-kamvar2014" class="csl-entry">
Kamvar, Zhian N., Javier F. Tabima, and Niklaus J. Grünwald. 2014. <span>“<span><em>Poppr</em></span>: An R Package for Genetic Analysis of Populations with Clonal, Partially Clonal, and/or Sexual Reproduction.”</span> <em>PeerJ</em> 2 (March): e281. <a href="https://doi.org/10.7717/peerj.281">https://doi.org/10.7717/peerj.281</a>.
</div>
<div id="ref-kang2010" class="csl-entry">
Kang, Hyun Min, Jae Hoon Sul, Susan K Service, Noah A Zaitlen, Sit-yee Kong, Nelson B Freimer, Chiara Sabatti, and Eleazar Eskin. 2010. <span>“Variance Component Model to Account for Sample Structure in Genome-Wide Association Studies.”</span> <em>Nature Genetics</em> 42 (4): 348–54. <a href="https://doi.org/10.1038/ng.548">https://doi.org/10.1038/ng.548</a>.
</div>
<div id="ref-Luu2017" class="csl-entry">
Luu, Keurcien, Eric Bazin, and Michael G. B. Blum. 2016. <span>“<span><em>Pcadapt</em></span>: An R Package to Perform Genome Scans for Selection Based on Principal Component Analysis.”</span> <em>Molecular Ecology Resources</em> 17 (1): 67–77. <a href="https://doi.org/10.1111/1755-0998.12592">https://doi.org/10.1111/1755-0998.12592</a>.
</div>
<div id="ref-paradis2018" class="csl-entry">
Paradis, Emmanuel, and Klaus Schliep. 2018. <span>“Ape 5.0: An Environment for Modern Phylogenetics and Evolutionary Analyses in R.”</span> Edited by Russell Schwartz. <em>Bioinformatics</em> 35 (3): 526–28. <a href="https://doi.org/10.1093/bioinformatics/bty633">https://doi.org/10.1093/bioinformatics/bty633</a>.
</div>
<div id="ref-whitlock2015" class="csl-entry">
Whitlock, Michael C., and Katie E. Lotterhos. 2015. <span>“Reliable Detection of Loci Responsible for Local Adaptation: Inference of a Null Model Through Trimming the Distribution of<span><em>F</em></span><span><sub>ST</sub></span>.”</span> <em>The American Naturalist</em> 186 (S1): S24–36. <a href="https://doi.org/10.1086/682949">https://doi.org/10.1086/682949</a>.
</div>
<div id="ref-yu2016" class="csl-entry">
Yu, Guangchuang, David K. Smith, Huachen Zhu, Yi Guan, and Tommy Tsan-Yuk Lam. 2016. <span>“Ggtree: An r Package for Visualization and Annotation of Phylogenetic Trees with Their Covariates and Other Associated Data.”</span> Edited by Greg McInerny. <em>Methods in Ecology and Evolution</em> 8 (1): 28–36. <a href="https://doi.org/10.1111/2041-210x.12628">https://doi.org/10.1111/2041-210x.12628</a>.
</div>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="sec-ai-report.html"><span class="header-section-number">8</span> AI Report</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#sec-index"><span class="header-section-number">9</span> FAQ</a></li>
<li><a class="nav-link" href="#general-usage">General Usage</a></li>
<li><a class="nav-link" href="#analysis-features">Analysis &amp; Features</a></li>
<li><a class="nav-link" href="#support">Support</a></li>
<li><a class="nav-link" href="#installation-issues">Installation Issues</a></li>
<li><a class="nav-link" href="#index">Index</a></li>
<li><a class="nav-link" href="#bibliography">Bibliography</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/TeddYenn/ShiNyP-guide/blob/master/10-Index.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/TeddYenn/ShiNyP-guide/edit/master/10-Index.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>User Guide for ShiNyP</strong>: SNP Analysis and Visualization Platform" was written by Yen-Hsiang Huang, Chung-Feng Kao. It was last built on May 2025.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
